<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Covariance Correlations</title>

<script src="site_libs/header-attrs-2.23/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">dgrp-starve</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/nklimko/dgrp-starve">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Covariance Correlations</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-09-05
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>dgrp-starve/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221101code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20221101)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221101code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221101)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/data2/morgante_lab/nklimko/rep/dgrp-starve/
</td>
<td style="text-align:left;">
.
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomnklimkodgrpstarvetreed41213f2a58512c11a14eae70d9bd4d51e06a702targetblankd41213fa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/nklimko/dgrp-starve/tree/d41213f2a58512c11a14eae70d9bd4d51e06a702" target="_blank">d41213f</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomnklimkodgrpstarvetreed41213f2a58512c11a14eae70d9bd4d51e06a702targetblankd41213fa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/nklimko/dgrp-starve/tree/d41213f2a58512c11a14eae70d9bd4d51e06a702" target="_blank">d41213f</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    code/snake/
    Ignored:    data/snake/
    Ignored:    junk/
    Ignored:    notes/

Untracked files:
    Untracked:  .snakemake/
    Untracked:  Rplot.pdf
    Untracked:  analysis/Method/
    Untracked:  analysis/linearReg.Rmd
    Untracked:  bglr-f.R
    Untracked:  cnn.R
    Untracked:  code/PCA/
    Untracked:  code/data-prep/
    Untracked:  code/fabio/
    Untracked:  code/gBayesC.R
    Untracked:  code/id_bank_creation.R
    Untracked:  code/intro-starve/
    Untracked:  code/methodComp/
    Untracked:  code/regress/
    Untracked:  colorCode
    Untracked:  data/all_hist_data_f.Rds
    Untracked:  data/all_hist_data_m.Rds
    Untracked:  data/bayesF.rds
    Untracked:  data/bayesM.rds
    Untracked:  data/bglr-f-130k.rds
    Untracked:  data/bglr-f.rds
    Untracked:  data/bglr-m-130k.rds
    Untracked:  data/bglr-m.rds
    Untracked:  data/corLoop-f-minus.rds
    Untracked:  data/corLoop-f.rds
    Untracked:  data/corLoop-m-Minus.rds
    Untracked:  data/corLoop-m-minus.rds
    Untracked:  data/corLoop-m.rds
    Untracked:  data/fRegress.txt
    Untracked:  data/fRegress_adj.txt
    Untracked:  data/fm.burglar
    Untracked:  data/gbayesC-f.Rds
    Untracked:  data/gbayesC-m.Rds
    Untracked:  data/gbayesC.Rds
    Untracked:  data/gbayes_100k-f.Rds
    Untracked:  data/gbayes_100k-m.Rds
    Untracked:  data/goGroups.txt
    Untracked:  data/hist_data_f.Rds
    Untracked:  data/hist_data_m.Rds
    Untracked:  data/id_bank
    Untracked:  data/id_bank.Rds
    Untracked:  data/mPart.txt
    Untracked:  data/mRegress.txt
    Untracked:  data/mRegress_adj.txt
    Untracked:  data/multiReg.rData
    Untracked:  data/pheno_f
    Untracked:  data/pheno_m
    Untracked:  data/pls_hist_data_f.Rds
    Untracked:  data/sr_hist_data_f.Rds
    Untracked:  data/sr_hist_data_m.Rds
    Untracked:  data/starve-f.txt
    Untracked:  data/starve-m.txt
    Untracked:  data/xp-f.txt
    Untracked:  data/xp-m.txt
    Untracked:  data/xp_f
    Untracked:  data/xp_m
    Untracked:  data/y_save.txt
    Untracked:  f-cor.png
    Untracked:  figure/
    Untracked:  m-cor.png
    Untracked:  posterPlots.R
    Untracked:  runtime.png
    Untracked:  snake/
    Untracked:  temp.R

Unstaged changes:
    Modified:   .Rprofile
    Modified:   .gitattributes
    Modified:   .gitignore
    Modified:   README.md
    Modified:   _workflowr.yml
    Modified:   analysis/_site.yml
    Modified:   analysis/about.Rmd
    Deleted:    analysis/allHist2.Rmd
    Deleted:    analysis/gremlo.R
    Modified:   analysis/license.Rmd
    Modified:   analysis/linReg.Rmd
    Deleted:    analysis/methodComp-f.Rmd
    Deleted:    analysis/methodComp-m.Rmd
    Modified:   analysis/methodComp.Rmd
    Modified:   analysis/methodPred.Rmd
    Modified:   analysis/multiComp.Rmd
    Modified:   analysis/multiReg.Rmd
    Modified:   analysis/pca.Rmd
    Modified:   analysis/predict.Rmd
    Modified:   analysis/recap.Rmd
    Modified:   analysis/rewrite.Rmd
    Modified:   analysis/starve.Rmd
    Deleted:    analysis/stepwise-f.Rmd
    Deleted:    analysis/stepwise-m.Rmd
    Deleted:    analysis/testing.R
    Deleted:    analysis/tips.Rmd
    Modified:   analysis/top3.Rmd
    Modified:   analysis/trace.Rmd
    Modified:   code/README.md
    Deleted:    code/baseScript-lineComp.R
    Deleted:    code/combineSNP.R
    Deleted:    code/four-comp.76979.err
    Deleted:    code/four-comp.76979.out
    Deleted:    code/four-comp.sbatch
    Deleted:    code/fourLinePrep.R
    Deleted:    code/line_avgMinus.R
    Deleted:    code/line_avgPlus.R
    Deleted:    code/line_difMinus.R
    Deleted:    code/line_difPlus.R
    Deleted:    code/snpGene.R
    Deleted:    code/starveDataPrep.R
    Modified:   data/README.md
    Modified:   data/avgMinus-result.txt
    Modified:   data/avgMinus.txt
    Modified:   data/avgPlus-result.txt
    Modified:   data/avgPlus.txt
    Modified:   data/difMinus-result.txt
    Modified:   data/difMinus.txt
    Modified:   data/difPlus-result.txt
    Modified:   data/difPlus.txt
    Modified:   data/geneHits.txt
    Modified:   data/snpList.txt
    Modified:   data/starve.csv
    Modified:   dgrp-starve.Rproj
    Modified:   output/README.md
    Modified:   output/avgMinus-result.txt
    Modified:   output/avgPlus-result.txt
    Modified:   output/difMinus-result.txt
    Modified:   output/difPlus-result.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/allHist.Rmd</code>) and HTML (<code>docs/allHist.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/d41213f2a58512c11a14eae70d9bd4d51e06a702/analysis/allHist.Rmd" target="_blank">d41213f</a>
</td>
<td>
nklimko
</td>
<td>
2023-09-05
</td>
<td>
wflow_publish(“analysis/allHist.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nklimko/dgrp-starve/fc8a5e4be5e219797a5dab246d57de0ad6e2399c/docs/allHist.html" target="_blank">fc8a5e4</a>
</td>
<td>
nklimko
</td>
<td>
2023-09-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nklimko/dgrp-starve/cd946a5381a72961f3a7278011a277976f5865d9/docs/allHist.html" target="_blank">cd946a5</a>
</td>
<td>
nklimko
</td>
<td>
2023-09-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/28b05db54e92ee8bc35650b52ee1fd4d414b1f31/analysis/allHist.Rmd" target="_blank">28b05db</a>
</td>
<td>
nklimko
</td>
<td>
2023-09-04
</td>
<td>
wflow_publish(“analysis/allHist.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nklimko/dgrp-starve/feaac67cfc1a09c2360625ce812ec1d4c282848f/docs/allHist.html" target="_blank">feaac67</a>
</td>
<td>
nklimko
</td>
<td>
2023-09-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/1606d88e6c802c82ae21ced332be356c814c5e28/analysis/allHist.Rmd" target="_blank">1606d88</a>
</td>
<td>
nklimko
</td>
<td>
2023-09-04
</td>
<td>
wflow_publish(“analysis/allHist.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nklimko/dgrp-starve/02ba6ed6a6eb26d74d7ad75facb5923ec69b08c0/docs/allHist.html" target="_blank">02ba6ed</a>
</td>
<td>
nklimko
</td>
<td>
2023-08-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/741e81d106c330cf681f753bbabf1185aa129c39/analysis/allHist.Rmd" target="_blank">741e81d</a>
</td>
<td>
nklimko
</td>
<td>
2023-08-29
</td>
<td>
wflow_publish(“analysis/allHist.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nklimko/dgrp-starve/2657f5c6dc6c600a1bde7dbcd1643f43281115ce/docs/allHist.html" target="_blank">2657f5c</a>
</td>
<td>
nklimko
</td>
<td>
2023-08-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/2f076f141ffc61b3817a6178847c19109aa38c53/analysis/allHist.Rmd" target="_blank">2f076f1</a>
</td>
<td>
nklimko
</td>
<td>
2023-08-29
</td>
<td>
wflow_publish(“analysis/allHist.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nklimko/dgrp-starve/882cd4930eb2907c53bd895d930f9cf88cd69b2e/docs/allHist.html" target="_blank">882cd49</a>
</td>
<td>
nklimko
</td>
<td>
2023-08-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/8ead827b248524e68d53af8c4d3816b17516016a/analysis/allHist.Rmd" target="_blank">8ead827</a>
</td>
<td>
nklimko
</td>
<td>
2023-08-29
</td>
<td>
allhist pseudo
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nklimko/dgrp-starve/9774b2a0a7f97c04748a9d26d1074c270b54fff6/docs/allHist.html" target="_blank">9774b2a</a>
</td>
<td>
nklimko
</td>
<td>
2023-08-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nklimko/dgrp-starve/debb4cd946ed4c087c44df3187863fa86686f5bc/docs/allHist.html" target="_blank">debb4cd</a>
</td>
<td>
nklimko
</td>
<td>
2023-08-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nklimko/dgrp-starve/d7f91559af7cae9d4ca10c8128c6785be86b1a13/docs/allHist.html" target="_blank">d7f9155</a>
</td>
<td>
nklimko
</td>
<td>
2023-08-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/2c78f8a3a7f47f912a5c6c5e45d5677601186040/analysis/allHist.Rmd" target="_blank">2c78f8a</a>
</td>
<td>
nklimko
</td>
<td>
2023-08-25
</td>
<td>
wflow_publish(“analysis/allHist.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nklimko/dgrp-starve/a64ae8125d1ddc9eb898d6f630fbb0abc54190a6/docs/allHist.html" target="_blank">a64ae81</a>
</td>
<td>
nklimko
</td>
<td>
2023-08-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/3505b0065dcea584620a31b709f22b34014b3e2a/analysis/allHist.Rmd" target="_blank">3505b00</a>
</td>
<td>
nklimko
</td>
<td>
2023-08-15
</td>
<td>
wflow_publish(“analysis/allHist.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>#Correlation Coefficient Paths
uniF &lt;- readRDS(&#39;snake/data/sr/40_all/f/cor/all.Rds&#39;)
uniM &lt;- readRDS(&#39;snake/data/sr/40_all/f/cor/all.Rds&#39;)

multiF &lt;- readRDS(&#39;snake/data/top3/40_all/f/cor/all.Rds&#39;)
multiM &lt;- readRDS(&#39;snake/data/top3/40_all/f/cor/all.Rds&#39;)



#Runtimes
timeF &lt;- &#39;snake/data/sr/40_all/f/time/all.Rds&#39;
timeM &lt;- &#39;snake/data/sr/40_all/f/time/all.Rds&#39;</code></pre>
<pre class="r"><code>#reorders 50 X m table to a (50*m) X 2 table 
#converts method into factor to retain order
ggTidy &lt;- function(data){

  for(i in 1:dim(data)[2]){
    
    name &lt;- colnames(data)[i]
    temp &lt;- cbind(rep(name, dim(data)[1]), data[,i, with=FALSE])
    
    if(i==1){
      hold &lt;- temp
    } else{
      hold &lt;- rbind(hold, temp, use.names=FALSE)
    }
  }
  colnames(hold) &lt;- c(&quot;method&quot;, &quot;cor&quot;)
  hold$method &lt;- factor(hold$method, levels=unique(hold$method))
  
  
  return(hold)
}

#wrapper for ggplot call to custom fill sex, title, and y axis label
ggMake &lt;- function(data, sex, custom.title, custom.Ylab){
  
  plothole &lt;- ggplot(data, aes(x=method, y=cor, fill=method)) +
    geom_violin(color = NA, width = 0.65) +
    geom_boxplot(color=&#39;#440154FF&#39;, width = 0.15) +
    theme_minimal() +
    stat_summary(fun=mean, color=&#39;#440154FF&#39;, geom=&#39;point&#39;, 
                 shape=18, size=3, show.legend=FALSE) +
    labs(x=NULL,y=custom.Ylab, tag=sex, title=custom.title) +
    theme(legend.position=&#39;none&#39;,
          axis.text.x = element_text(angle = -45, size=10),
          text=element_text(size=10),
          plot.tag = element_text(size=15)) +
    scale_fill_viridis(begin = 0.4, end=0.9,discrete=TRUE)
  
  return(plothole)
  
}</code></pre>
<div id="intro-to-why" class="section level3">
<h3>Intro to why</h3>
<p>Complex trait prediction is based on the notion that some traits are impacted by a large number of small-effect genes which makes diagnosis or prediction challenging.</p>
<p>We are using gene expression data over genomic frequencies as they have shown previously to provide more accurate models.</p>
<p>This project is designed to establish a baseline comparison for statistical methods commonly used in the field.</p>
</div>
<div id="intro-to-methods" class="section level3">
<h3>Intro to methods</h3>
<p>In order to improve model accuracy, we look at shrinkage and variable selection. Too much of either can result in overfitting of the model, but failure to perform either results in large variance that diminishes model accuracy.</p>
<p>Shrinkage is based on the assumption that not all independent variables have the same effect on the response variable in a given model. The effects are scaled by a distribution of our choosing based on our assumption of the model. A normal distribution centered around 0 will scale most effects down while leaving more significant effects closer to 1. Changing the shape of the distribution changes the degree of shrinkage applied to the model.</p>
<p>Variable selection takes another step in model assumptions. We assume under a general model that all variables have some effect on the response even if the effect is small. In order to reduce noise, we can select for variables that only have stronger effects by discarding weaker or insignificant independent variables from the model. With genes, this models our assumption that not all genes have an effect on the trait of interest, while a number of genes will have a pronounced effect on our complex trait.</p>
</div>
<div id="method-line-by-line" class="section level3">
<h3>Method line by line</h3>
<ul>
<li><p>Random Forest: Random forest is a machine learning method that creates random networks of decision trees to create a linear regression Random Forest decision trees are used to discover impactful genes similar to how PCR selects for variance maximizing components.</p></li>
<li><p>Principal Component Regression (PCR): Principal Component Regression uses principal component analysis to find combinations of independent variables that maximize the change in variance on the response variable. This process greatly reduces the dimensionality of the data. Within this, variable selection can be performed optimally by finding the greatest change in total variance explained and only using principal components up to that point.</p></li>
<li><p>Neural Network (NN): Neural networks are a type of machine learning method where hidden layers of hyperparameters are used to evaluate the data. Learning is rewarded by maximizing change in variance. The current model has five layers with 5000, 2500, and 1000 hyperparameters in order.</p></li>
<li><p>Bayesian Variable Selection (BVS): Bayesian Variable Selection(BVS) uses a spike-and-slab prior to perform effect shrinkage and variable selection. The model is fit using Variational Inference(VI) methods.</p></li>
<li><p>BayesC - BayesC uses a spike-and-slab prior to perform effect shrinkage and variable selection. The model is fit using Markov Chain Monte Carlo(MCMC) methods.</p></li>
<li><p>Genomic Best Linear Unbiased Predictor (GBLUP): GBLUP is a linear mixed model created using the covariance of genomic relationships(geneXgene) to perform effect shrinkage.</p></li>
<li><p>Ridge Regression (RR):Ridge Regression is a penalized regression method that performs effect shrinkage only.</p></li>
<li><p>Least Absolute Shrinkage and Selection Operator (LASSO): LASSO uses the same penalized regression as Ridge regression with an added parameter to perform both effect shrinkage and variable selection.</p></li>
<li><p>Multiple Regression with Multivariate Adaptive Shrinkage (MR.MASH): Mr. Mash performs shrinkage and variable selection by using a custom mixed prior. This prior can be generated using summary statistics alone or can be layered with covariance matrices derived from strong effects.</p></li>
</ul>
</div>
<div id="all-models" class="section level3">
<h3>All Models</h3>
<p>Wtih traits that are correlated, we expect that changes in the trait of interest can be described by changes in additional trait values. We sought to incorporate this by taking the top three correlated traits into account in our multivariate models. The traits capillary feeding, free glycerol, and free glucose were selected for both males and females.</p>
<p>In order to see if correlated traits could be used to improve model accuracy, we performed both univariate and multivariate analyses with models that permitted multiple response prediction.</p>
<p>For every method tested, fifty replicates were created using an 80% train/20% test data split by DGRP2 line. Each column below is a distribution of the method’s correlation coefficient from all replicates.</p>
<pre class="r"><code>#All Female
allData &lt;- data.table(uniF, multiF)
print(colMeans(na.omit(allData)))</code></pre>
<pre><code>         rf      bayesC       gblup         pcr          nn      mr.ash 
 0.35128325  0.30140171  0.30374352  0.04416158  0.05540557  0.18472388 
         rr       lasso         pls   datadrive      mlasso     mr.mash 
 0.28260545  0.39068512  0.35476129  0.12601118  0.19058669  0.09047196 
 multigblup multibayesC 
 0.31083037  0.28933714 </code></pre>
<pre class="r"><code>data &lt;- na.omit(ggTidy(allData))
gg[[1]] &lt;- ggMake(data, &#39;F&#39;, &#39;All Female&#39;, &#39;Correlation&#39;)

#Univariate Female
uniData &lt;- uniF
data &lt;- na.omit(ggTidy(uniData))
gg[[3]] &lt;- ggMake(data, &#39;F&#39;, &#39;Univariate Female&#39;, &#39;Correlation&#39;)

#All Male
allData &lt;- data.table(uniM, multiM)
print(colMeans(na.omit(allData)))</code></pre>
<pre><code>         rf      bayesC       gblup         pcr          nn      mr.ash 
 0.35128325  0.30140171  0.30374352  0.04416158  0.05540557  0.18472388 
         rr       lasso         pls   datadrive      mlasso     mr.mash 
 0.28260545  0.39068512  0.35476129  0.12601118  0.19058669  0.09047196 
 multigblup multibayesC 
 0.31083037  0.28933714 </code></pre>
<pre class="r"><code>data &lt;- na.omit(ggTidy(allData))
gg[[2]] &lt;- ggMake(data, &#39;M&#39;, &#39;All Male&#39;, &#39;Correlation&#39;)

#Univariate Female
uniData &lt;- uniM
data &lt;- na.omit(ggTidy(uniData))
gg[[4]] &lt;- ggMake(data, &#39;M&#39;, &#39;Univariate Male&#39;, &#39;Correlation&#39;)</code></pre>
</div>
<div id="univariate-models" class="section level3">
<h3>Univariate Models</h3>
<pre class="r"><code>plot_grid(gg[[1]], ncol=1)</code></pre>
<p><img src="figure/allHist.Rmd/plot-print-all-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-print-all-1">
Past versions of plot-print-all-1.png
</button>
</p>
<div id="fig-plot-print-all-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/cd946a5381a72961f3a7278011a277976f5865d9/docs/figure/allHist.Rmd/plot-print-all-1.png" target="_blank">cd946a5</a>
</td>
<td>
nklimko
</td>
<td>
2023-09-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/feaac67cfc1a09c2360625ce812ec1d4c282848f/docs/figure/allHist.Rmd/plot-print-all-1.png" target="_blank">feaac67</a>
</td>
<td>
nklimko
</td>
<td>
2023-09-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/02ba6ed6a6eb26d74d7ad75facb5923ec69b08c0/docs/figure/allHist.Rmd/plot-print-all-1.png" target="_blank">02ba6ed</a>
</td>
<td>
nklimko
</td>
<td>
2023-08-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/2657f5c6dc6c600a1bde7dbcd1643f43281115ce/docs/figure/allHist.Rmd/plot-print-all-1.png" target="_blank">2657f5c</a>
</td>
<td>
nklimko
</td>
<td>
2023-08-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/882cd4930eb2907c53bd895d930f9cf88cd69b2e/docs/figure/allHist.Rmd/plot-print-all-1.png" target="_blank">882cd49</a>
</td>
<td>
nklimko
</td>
<td>
2023-08-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_grid(gg[[2]], ncol=1)</code></pre>
<p><img src="figure/allHist.Rmd/plot-print-all-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-print-all-2">
Past versions of plot-print-all-2.png
</button>
</p>
<div id="fig-plot-print-all-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/cd946a5381a72961f3a7278011a277976f5865d9/docs/figure/allHist.Rmd/plot-print-all-2.png" target="_blank">cd946a5</a>
</td>
<td>
nklimko
</td>
<td>
2023-09-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/feaac67cfc1a09c2360625ce812ec1d4c282848f/docs/figure/allHist.Rmd/plot-print-all-2.png" target="_blank">feaac67</a>
</td>
<td>
nklimko
</td>
<td>
2023-09-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/02ba6ed6a6eb26d74d7ad75facb5923ec69b08c0/docs/figure/allHist.Rmd/plot-print-all-2.png" target="_blank">02ba6ed</a>
</td>
<td>
nklimko
</td>
<td>
2023-08-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/2657f5c6dc6c600a1bde7dbcd1643f43281115ce/docs/figure/allHist.Rmd/plot-print-all-2.png" target="_blank">2657f5c</a>
</td>
<td>
nklimko
</td>
<td>
2023-08-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/882cd4930eb2907c53bd895d930f9cf88cd69b2e/docs/figure/allHist.Rmd/plot-print-all-2.png" target="_blank">882cd49</a>
</td>
<td>
nklimko
</td>
<td>
2023-08-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Comparison shows that multivariate methods do not significantly improve the model accuracy. As such, we will focus on univariate methods moving forward.</p>
<p>Below are the univariate models alone.</p>
<pre class="r"><code>plot_grid(gg[[3]], ncol=1)</code></pre>
<p><img src="figure/allHist.Rmd/plot-print-univariate-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-print-univariate-1">
Past versions of plot-print-univariate-1.png
</button>
</p>
<div id="fig-plot-print-univariate-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/cd946a5381a72961f3a7278011a277976f5865d9/docs/figure/allHist.Rmd/plot-print-univariate-1.png" target="_blank">cd946a5</a>
</td>
<td>
nklimko
</td>
<td>
2023-09-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/feaac67cfc1a09c2360625ce812ec1d4c282848f/docs/figure/allHist.Rmd/plot-print-univariate-1.png" target="_blank">feaac67</a>
</td>
<td>
nklimko
</td>
<td>
2023-09-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/02ba6ed6a6eb26d74d7ad75facb5923ec69b08c0/docs/figure/allHist.Rmd/plot-print-univariate-1.png" target="_blank">02ba6ed</a>
</td>
<td>
nklimko
</td>
<td>
2023-08-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/2657f5c6dc6c600a1bde7dbcd1643f43281115ce/docs/figure/allHist.Rmd/plot-print-univariate-1.png" target="_blank">2657f5c</a>
</td>
<td>
nklimko
</td>
<td>
2023-08-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/882cd4930eb2907c53bd895d930f9cf88cd69b2e/docs/figure/allHist.Rmd/plot-print-univariate-1.png" target="_blank">882cd49</a>
</td>
<td>
nklimko
</td>
<td>
2023-08-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_grid(gg[[4]], ncol=1)</code></pre>
<p><img src="figure/allHist.Rmd/plot-print-univariate-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-print-univariate-2">
Past versions of plot-print-univariate-2.png
</button>
</p>
<div id="fig-plot-print-univariate-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/cd946a5381a72961f3a7278011a277976f5865d9/docs/figure/allHist.Rmd/plot-print-univariate-2.png" target="_blank">cd946a5</a>
</td>
<td>
nklimko
</td>
<td>
2023-09-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/feaac67cfc1a09c2360625ce812ec1d4c282848f/docs/figure/allHist.Rmd/plot-print-univariate-2.png" target="_blank">feaac67</a>
</td>
<td>
nklimko
</td>
<td>
2023-09-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/02ba6ed6a6eb26d74d7ad75facb5923ec69b08c0/docs/figure/allHist.Rmd/plot-print-univariate-2.png" target="_blank">02ba6ed</a>
</td>
<td>
nklimko
</td>
<td>
2023-08-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/2657f5c6dc6c600a1bde7dbcd1643f43281115ce/docs/figure/allHist.Rmd/plot-print-univariate-2.png" target="_blank">2657f5c</a>
</td>
<td>
nklimko
</td>
<td>
2023-08-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/882cd4930eb2907c53bd895d930f9cf88cd69b2e/docs/figure/allHist.Rmd/plot-print-univariate-2.png" target="_blank">882cd49</a>
</td>
<td>
nklimko
</td>
<td>
2023-08-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Rocky Linux 8.5 (Green Obsidian)

Matrix products: default
BLAS/LAPACK: /opt/ohpc/pub/libs/gnu9/openblas/0.3.7/lib/libopenblasp-r0.3.7.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] reshape2_1.4.4    melt_1.10.0       ggcorrplot_0.1.4  lubridate_1.9.2  
 [5] forcats_1.0.0     stringr_1.5.0     purrr_1.0.1       readr_2.1.4      
 [9] tidyr_1.3.0       tibble_3.2.1      tidyverse_2.0.0   scales_1.2.1     
[13] viridis_0.6.2     viridisLite_0.4.2 qqman_0.1.8       cowplot_1.1.1    
[17] ggplot2_3.4.2     data.table_1.14.8 dplyr_1.1.2       workflowr_1.7.0  

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.11       getPass_0.2-2     ps_1.7.5          rprojroot_2.0.3  
 [5] digest_0.6.33     utf8_1.2.3        plyr_1.8.8        R6_2.5.1         
 [9] evaluate_0.21     httr_1.4.5        highr_0.10        pillar_1.9.0     
[13] rlang_1.1.1       rstudioapi_0.15.0 whisker_0.4.1     callr_3.7.3      
[17] jquerylib_0.1.4   rmarkdown_2.23    labeling_0.4.2    munsell_0.5.0    
[21] compiler_4.1.2    httpuv_1.6.9      xfun_0.39         pkgconfig_2.0.3  
[25] htmltools_0.5.5   tidyselect_1.2.0  gridExtra_2.3     fansi_1.0.4      
[29] calibrate_1.7.7   tzdb_0.3.0        withr_2.5.0       later_1.3.1      
[33] MASS_7.3-60       grid_4.1.2        jsonlite_1.8.7    gtable_0.3.3     
[37] lifecycle_1.0.3   git2r_0.31.0      magrittr_2.0.3    cli_3.6.1        
[41] stringi_1.7.12    cachem_1.0.8      farver_2.1.1      fs_1.6.3         
[45] promises_1.2.0.1  bslib_0.5.0       generics_0.1.3    vctrs_0.6.2      
[49] tools_4.1.2       glue_1.6.2        hms_1.1.3         processx_3.8.2   
[53] fastmap_1.1.1     yaml_2.3.7        timechange_0.2.0  colorspace_2.1-0 
[57] knitr_1.43        sass_0.4.7       </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
