<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Linear Regression - Adjusted Phenotypes+</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">dgrp-starve</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/nklimko/dgrp-starve">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Linear Regression - Adjusted Phenotypes+</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-01-22
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>dgrp-starve/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div class="panel-group" id="workflowr-checks">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221101code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20221101)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221101code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221101)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/data/morgante_lab/nklimko/rep/dgrp-starve/data/fRegress_adj.txt
</td>
<td style="text-align:left;">
data/fRegress_adj.txt
</td>
</tr>
<tr>
<td style="text-align:left;">
/data/morgante_lab/nklimko/rep/dgrp-starve/data/f_adj.txt
</td>
<td style="text-align:left;">
data/f_adj.txt
</td>
</tr>
<tr>
<td style="text-align:left;">
/data/morgante_lab/nklimko/rep/dgrp-starve/data/mRegress_adj.txt
</td>
<td style="text-align:left;">
data/mRegress_adj.txt
</td>
</tr>
<tr>
<td style="text-align:left;">
/data/morgante_lab/nklimko/rep/dgrp-starve/data/m_adj.txt
</td>
<td style="text-align:left;">
data/m_adj.txt
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomnklimkodgrpstarvetreeaee9a185dc4b087896b0ba78aac0ae5f22b7a28etargetblankaee9a18a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/nklimko/dgrp-starve/tree/aee9a185dc4b087896b0ba78aac0ae5f22b7a28e" target="_blank">aee9a18</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomnklimkodgrpstarvetreeaee9a185dc4b087896b0ba78aac0ae5f22b7a28etargetblankaee9a18a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/nklimko/dgrp-starve/tree/aee9a185dc4b087896b0ba78aac0ae5f22b7a28e" target="_blank">aee9a18</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData

Untracked files:
    Untracked:  analysis/gremlo.R
    Untracked:  analysis/linearReg.Rmd
    Untracked:  analysis/predict.Rmd
    Untracked:  analysis/rewrite.Rmd
    Untracked:  code/aaaTest
    Untracked:  code/analysisSR.R
    Untracked:  code/geneGO.R
    Untracked:  code/multiPrep.R
    Untracked:  code/regress.81916.err
    Untracked:  code/regress.81916.out
    Untracked:  code/regress.81918.err
    Untracked:  code/regress.81918.out
    Untracked:  code/regress.R
    Untracked:  code/regress.sbatch
    Untracked:  code/regressF.81919.err
    Untracked:  code/regressF.81919.out
    Untracked:  code/regressF.R
    Untracked:  code/regressF.sbatch
    Untracked:  code/regress_f_adj.109973.err
    Untracked:  code/regress_f_adj.109973.out
    Untracked:  code/regress_f_adj.109974.err
    Untracked:  code/regress_f_adj.109974.out
    Untracked:  code/regress_f_adj.R
    Untracked:  code/regress_f_adj.sbatch
    Untracked:  code/regress_m_adj.109971.err
    Untracked:  code/regress_m_adj.109971.out
    Untracked:  code/regress_m_adj.109972.err
    Untracked:  code/regress_m_adj.109972.out
    Untracked:  code/regress_m_adj.R
    Untracked:  code/regress_m_adj.sbatch
    Untracked:  code/snpGene.77509.err
    Untracked:  code/snpGene.77509.out
    Untracked:  code/snpGene.77515.err
    Untracked:  code/snpGene.77515.out
    Untracked:  code/snpGene.sbatch
    Untracked:  data/eQTL_traits_females.csv
    Untracked:  data/eQTL_traits_males.csv
    Untracked:  data/fMeans.txt
    Untracked:  data/fRegress.txt
    Untracked:  data/fRegress_adj.txt
    Untracked:  data/f_adj.txt
    Untracked:  data/goGroups.txt
    Untracked:  data/mMeans.txt
    Untracked:  data/mPart.txt
    Untracked:  data/mRegress.txt
    Untracked:  data/mRegress_adj.txt
    Untracked:  data/m_adj.txt
    Untracked:  data/multiReg.rData
    Untracked:  data/starve-f.txt
    Untracked:  data/starve-m.txt
    Untracked:  data/xp-f.txt
    Untracked:  data/xp-m.txt
    Untracked:  data/y_save.txt
    Untracked:  figure/
    Untracked:  lmm.R
    Untracked:  qqdum.R
    Untracked:  scoreAnalysisMulticomp.R
    Untracked:  temp.Rmd

Unstaged changes:
    Deleted:    analysis/database.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/linReg.Rmd
    Modified:   analysis/multiReg.Rmd
    Modified:   analysis/recap.Rmd
    Deleted:    analysis/scripts.Rmd
    Modified:   code/baseScript-lineComp.R
    Modified:   code/fourLinePrep.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="prep" class="section level1">
<h1>Prep</h1>
<pre class="r"><code>#load mgl data and custom function
load(&quot;/data/morgante_lab/data/dgrp/misc/adjustData.RData&quot;)

#read in expression data
f_data &lt;- fread(&quot;data/fMeans.txt&quot;)

#grab starvation
y &lt;- f_data[,starvation]

#store starvation with id
dat &lt;- data.frame(id=f_data[,line], y=y)

#adjust data with given function
f_adjPheno &lt;- adjustPheno(dat, &quot;starvation&quot;)</code></pre>
<pre><code>Type III ANOVA table for covariates: starvation
                 Df Sum of Sq   RSS    AIC F value  Pr(&gt;F)  
&lt;none&gt;                        28767 1009.8                  
factor(wolba)     1    483.10 29250 1011.1  3.1236 0.07881 .
factor(In_2L_t)   2     60.22 28827 1006.2  0.1947 0.82325  
factor(In_2R_NS)  2    300.22 29067 1007.8  0.9706 0.38078  
factor(In_3R_P)   2     85.92 28853 1006.4  0.2778 0.75777  
factor(In_3R_K)   2    959.43 29726 1012.3  3.1017 0.04731 *
factor(In_3R_Mo)  2    416.65 29184 1008.6  1.3470 0.26255  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1


Estimated effects
                    Estimate Std. Error    t value     Pr(&gt;|t|)
(Intercept)       58.7465186   1.553929 37.8051442 3.103130e-89
factor(wolba)y     3.2450739   1.836101  1.7673724 7.880567e-02
factor(In_2L_t)1  -0.8506378   3.101713 -0.2742477 7.841986e-01
factor(In_2L_t)2  -1.8583916   3.151661 -0.5896547 5.561378e-01
factor(In_2R_NS)1  0.8630221   4.621570  0.1867379 8.520697e-01
factor(In_2R_NS)2  6.7057598   4.825270  1.3897169 1.662760e-01
factor(In_3R_P)1  -1.8513843   5.206148 -0.3556150 7.225319e-01
factor(In_3R_P)2   4.0596759   6.331413  0.6411959 5.221847e-01
factor(In_3R_K)1   7.4870256   4.084124  1.8332024 6.837092e-02
factor(In_3R_K)2  15.5762633   8.948152  1.7407240 8.338564e-02
factor(In_3R_Mo)1 -5.4712766   4.279569 -1.2784644 2.026787e-01
factor(In_3R_Mo)2 -3.6074434   3.224295 -1.1188318 2.646547e-01</code></pre>
<pre class="r"><code>#overwrite starvation column
f_adj &lt;- f_data[,starvation:=f_adjPheno]

#save to txt file
fwrite(f_adj, &quot;data/f_adj.txt&quot;)</code></pre>
<pre class="r"><code>#load mgl data and custom function
load(&quot;/data/morgante_lab/data/dgrp/misc/adjustData.RData&quot;)

#read in expression data
m_data &lt;- fread(&quot;data/mMeans.txt&quot;)

#grab starvation
y &lt;- m_data[,starvation]

#store starvation with id
dat &lt;- data.frame(id=m_data[,line], y=y)

#adjust data with given function
m_adjPheno &lt;- adjustPheno(dat, &quot;starvation&quot;)</code></pre>
<pre><code>Type III ANOVA table for covariates: starvation
                 Df Sum of Sq   RSS    AIC F value Pr(&gt;F)
&lt;none&gt;                        15975 893.32               
factor(wolba)     1     3.041 15978 891.35  0.0354 0.8509
factor(In_2L_t)   2   299.141 16274 892.99  1.7415 0.1781
factor(In_2R_NS)  2   243.044 16218 892.31  1.4149 0.2455
factor(In_3R_P)   2   230.105 16205 892.15  1.3396 0.2645
factor(In_3R_K)   2   288.050 16263 892.85  1.6770 0.1898
factor(In_3R_Mo)  2   207.445 16182 891.87  1.2077 0.3012


Estimated effects
                    Estimate Std. Error    t value     Pr(&gt;|t|)
(Intercept)       45.6614374   1.157976 39.4321227 2.909220e-92
factor(wolba)y    -0.2574606   1.368248 -0.1881681 8.509500e-01
factor(In_2L_t)1   1.7198329   2.311372  0.7440744 4.577704e-01
factor(In_2L_t)2  -3.7780449   2.348592 -1.6086422 1.093906e-01
factor(In_2R_NS)1 -1.0811449   3.443957 -0.3139252 7.539297e-01
factor(In_2R_NS)2  5.8594026   3.595753  1.6295343 1.048923e-01
factor(In_3R_P)1  -2.0731924   3.879580 -0.5343858 5.937128e-01
factor(In_3R_P)2   7.1994131   4.718118  1.5259077 1.287315e-01
factor(In_3R_K)1   5.3513957   3.043456  1.7583284 8.033635e-02
factor(In_3R_K)2   3.7673764   6.668091  0.5649857 5.727643e-01
factor(In_3R_Mo)1 -3.0569154   3.189101 -0.9585510 3.390293e-01
factor(In_3R_Mo)2 -3.0977753   2.402719 -1.2892793 1.989020e-01</code></pre>
<pre class="r"><code>#overwrite starvation column
m_adj &lt;- m_data[,starvation:=m_adjPheno]

#save to txt file
fwrite(m_adj, &quot;data/m_adj.txt&quot;)</code></pre>
</div>
<div id="correlations" class="section level1">
<h1>Correlations</h1>
<p>In order to save time, I used secretariat’s computational resources to perform the correlation calculations in parallel. The following two sections of code were submission scripts performed in secretariat.</p>
<pre class="r"><code>library(dplyr)
library(data.table)
library(doParallel)

registerDoParallel(cores = 12)

# file path for final table
filePath &lt;- &quot;/data/morgante_lab/nklimko/rep/dgrp-starve/data/fRegress_adj.txt&quot;

# gene expression data from earlier bound to line and starvation
f_adj &lt;- fread(&quot;/data/morgante_lab/nklimko/rep/dgrp-starve/data/f_adj.txt&quot;)

start &lt;- 3
end &lt;- 11340
#f 11340
#m 13577

# foreach creates a list with index indicating position in loop
# function grabs the p-value from correlation of starvation to expression of trait
pvalList &lt;- foreach(i=start:end) %dopar% {  
  temp &lt;- summary(lm(f_adj[,c(2,i), with=FALSE], na.action=na.omit))[[4]][8]  
}

# converts list to vector
part &lt;- unlist(pvalList, use.names = FALSE)

# binds p-vals to column index 
id &lt;- start:end
part &lt;- data.table(id,pvalList)

# write table to file
fwrite(part, filePath)</code></pre>
<p>Both of these are available by clicking the drop down buttons on the right -&gt;</p>
<pre class="r"><code>library(dplyr)
library(data.table)
library(doParallel)

registerDoParallel(cores = 12)

# file path for final table
filePath &lt;- &quot;/data/morgante_lab/nklimko/rep/dgrp-starve/data/mRegress_adj.txt&quot;

# gene expression data from earlier bound to line and starvation
m_adj &lt;- fread(&quot;/data/morgante_lab/nklimko/rep/dgrp-starve/data/m_adj.txt&quot;)

start &lt;- 3
end &lt;- 13577
#f 11340
#m 13577

# foreach creates a list with index indicating position in loop
# function grabs the p-value from correlation of starvation to expression of trait
pvalList &lt;- foreach(i=start:end) %dopar% {  
  temp &lt;- summary(lm(m_adj[,c(2,i), with=FALSE], na.action=na.omit))[[4]][8]  
}

# converts list to vector
part &lt;- unlist(pvalList, use.names = FALSE)

# binds p-vals to column index 
id &lt;- start:end
part &lt;- data.table(id,pvalList)

# write table to file
fwrite(part, filePath)</code></pre>
<div id="qq-and-manthattan-data-prep--" class="section level4">
<h4>QQ and Manthattan Data Prep -&gt;</h4>
<pre class="r"><code># Female qq prep

#f 11340
#m 13577

gg &lt;- vector(mode=&#39;list&#39;, length=4)

#read in results from compute node
fReg &lt;- fread(&quot;data/fRegress_adj.txt&quot;)

#number of data points
n &lt;- dim(fReg)[1]

#Theoretical Quantiles
uniform &lt;- 1:n/(n+1)

#sorted p values
empirical &lt;- sort(fReg[,pvalList])

#-log 10
logPlot &lt;- -log10(fReg[,pvalList])

#table for qqplot
qqdata &lt;- data.table(uniform, empirical, logPlot)

qqdataF &lt;- qqdata

# ggplot of data to fpr, qqplot
if(FALSE){
gg[[1]] &lt;- ggplot(qqdata, aes(x=uniform, y=empirical)) +
  geom_point(color=&quot;red&quot;) +  
  geom_segment(aes(x=0,y=0,xend=1,yend=1)) + 
  xlab(&quot;Theoretical Quantiles&quot;) +
  ylab(&quot;Sorted p-values&quot;) +
  ggtitle(&quot;Female Expression QQ Plot&quot;)
}

gg[[2]] &lt;- ggplot(qqdata, aes(x=uniform, y=logPlot)) +
  geom_point(color=&quot;red&quot;) + 
  geom_hline(yintercept= -log10(0.05/n), color=&quot;magenta&quot;) +
  xlab(&quot;Theoretical Quantiles&quot;) +
  ylab(&quot;-log10 p-values&quot;) +
  ggtitle(&quot;Female Expression Manhattan Plot&quot;)



## Code for looking at specific genes
#fMeans &lt;- fread(&quot;data/fMeans.txt&quot;)
#genes &lt;- colnames(fMeans)
#genes &lt;- genes[3:length(genes)]
#fReg &lt;- fReg[, gene:=genes]</code></pre>
<pre class="r"><code># Male qq prep

#read in results from compute node
mReg &lt;- fread(&quot;data/mRegress_adj.txt&quot;)

#number of data points
n &lt;- dim(mReg)[1]

#Theoretical Quantiles
uniform &lt;- 1:n/(n+1)

#sorted p values
empirical &lt;- sort(mReg[,pvalList])

#-log 10
logPlot &lt;- -log10(mReg[,pvalList])

#table for qqplot
qqdata &lt;- data.table(uniform, empirical, logPlot)

qqdataM &lt;- qqdata


# ggplot of data to fpr, qqplot
if(FALSE){
gg[[3]] &lt;- ggplot(qqdata, aes(x=uniform, y=empirical)) +
  geom_point(color=&quot;blue&quot;) +  
  geom_segment(aes(x=0,y=0,xend=1,yend=1), linewidth=1) + 
  xlab(&quot;Theoretical Quantiles&quot;) +
  ylab(&quot;Sorted p-values&quot;) +
  ggtitle(&quot;Male Expression QQ Plot&quot;)
}


gg[[4]] &lt;- ggplot(qqdata, aes(x=uniform, y=logPlot)) +
  geom_point(color=&quot;blue&quot;) + 
  geom_hline(yintercept= -log10(0.05/n), color=&quot;cyan&quot;) +
  xlab(&quot;Theoretical Quantiles&quot;) +
  ylab(&quot;-log10 p-values&quot;) +
  ggtitle(&quot;Male Expression Manhattan Plot&quot;)

## Code for looking at specific genes
#mMeans &lt;- fread(&quot;data/mMeans.txt&quot;)
#genes &lt;- colnames(mMeans)
#genes &lt;- genes[3:length(genes)]
#mReg &lt;- mReg[, gene:=genes]</code></pre>
</div>
</div>
<div id="qq-plots" class="section level1">
<h1>QQ Plots</h1>
<pre class="r"><code>par(mfrow=c(1,2))

qq(qqdataF[,empirical], main=&quot;Female Gene p-values&quot;)
qq(qqdataM[,empirical], main=&quot;Male Gene p-values&quot;)</code></pre>
<p><img src="figure/linearReg.Rmd/qq-plot-1.svg" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="manhattan-plots" class="section level1">
<h1>Manhattan Plots</h1>
<pre class="r"><code>par(mfrow=c(1,2))

plot_grid(gg[[2]],gg[[4]], ncol=2)</code></pre>
<p><img src="figure/linearReg.Rmd/manhattan-plot-1.svg" width="672" style="display: block; margin: auto;" /></p>
<p>Part of the reason both plots deviate from linearity because we did not account for genes that were correlated with each other.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /opt/ohpc/pub/Software/openblas_0.3.10/lib/libopenblas_haswellp-r0.3.10.dev.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] qqman_0.1.8       cowplot_1.1.1     ggplot2_3.3.5     data.table_1.14.2
[5] dplyr_1.0.8      

loaded via a namespace (and not attached):
 [1] tidyselect_1.1.2 xfun_0.30        bslib_0.3.1      purrr_0.3.4     
 [5] colorspace_2.0-3 vctrs_0.4.1      generics_0.1.2   htmltools_0.5.2 
 [9] yaml_2.3.5       utf8_1.2.2       rlang_1.0.4      jquerylib_0.1.4 
[13] later_1.3.0      pillar_1.7.0     glue_1.6.2       withr_2.5.0     
[17] DBI_1.1.2        calibrate_1.7.7  lifecycle_1.0.1  stringr_1.4.0   
[21] munsell_0.5.0    gtable_0.3.0     workflowr_1.7.0  evaluate_0.15   
[25] labeling_0.4.2   knitr_1.38       fastmap_1.1.0    httpuv_1.6.5    
[29] fansi_1.0.3      highr_0.9        Rcpp_1.0.8.3     promises_1.2.0.1
[33] scales_1.2.0     jsonlite_1.8.0   farver_2.1.0     fs_1.5.2        
[37] digest_0.6.29    stringi_1.7.6    rprojroot_2.0.3  grid_4.0.3      
[41] cli_3.3.0        tools_4.0.3      magrittr_2.0.3   sass_0.4.1      
[45] tibble_3.1.6     crayon_1.5.1     pkgconfig_2.0.3  ellipsis_0.3.2  
[49] MASS_7.3-56      assertthat_0.2.1 rmarkdown_2.16   rstudioapi_0.13 
[53] R6_2.5.1         git2r_0.30.1     compiler_4.0.3  </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
