<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Simple Regression</title>

<script src="site_libs/header-attrs-2.16/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">dgrp-starve</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/nklimko/dgrp-starve">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Simple Regression</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-01-08
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>dgrp-starve/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221101code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221101)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221101code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221101)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/data/morgante_lab/nklimko/rep/dgrp-starve/data/fRegress.txt
</td>
<td style="text-align:left;">
data/fRegress.txt
</td>
</tr>
<tr>
<td style="text-align:left;">
/data/morgante_lab/nklimko/rep/dgrp-starve/data/fMeans.txt
</td>
<td style="text-align:left;">
data/fMeans.txt
</td>
</tr>
<tr>
<td style="text-align:left;">
/data/morgante_lab/nklimko/rep/dgrp-starve/data/mRegress.txt
</td>
<td style="text-align:left;">
data/mRegress.txt
</td>
</tr>
<tr>
<td style="text-align:left;">
/data/morgante_lab/nklimko/rep/dgrp-starve/data/mMeans.txt
</td>
<td style="text-align:left;">
data/mMeans.txt
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomnklimkodgrpstarvetree951799a1f59b0f73117a1a9fdd6090b554058cfetargetblank951799aa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/nklimko/dgrp-starve/tree/951799a1f59b0f73117a1a9fdd6090b554058cfe" target="_blank">951799a</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomnklimkodgrpstarvetree951799a1f59b0f73117a1a9fdd6090b554058cfetargetblank951799aa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/nklimko/dgrp-starve/tree/951799a1f59b0f73117a1a9fdd6090b554058cfe" target="_blank">951799a</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  code/aaaTest
    Untracked:  code/analysisSR.R
    Untracked:  code/geneGO.R
    Untracked:  code/multiPrep.R
    Untracked:  code/regress.81916.err
    Untracked:  code/regress.81916.out
    Untracked:  code/regress.81918.err
    Untracked:  code/regress.81918.out
    Untracked:  code/regress.R
    Untracked:  code/regress.sbatch
    Untracked:  code/regressF.81919.err
    Untracked:  code/regressF.81919.out
    Untracked:  code/regressF.R
    Untracked:  code/regressF.sbatch
    Untracked:  code/snpGene.77509.err
    Untracked:  code/snpGene.77509.out
    Untracked:  code/snpGene.77515.err
    Untracked:  code/snpGene.77515.out
    Untracked:  code/snpGene.sbatch
    Untracked:  data/eQTL_traits_females.csv
    Untracked:  data/eQTL_traits_males.csv
    Untracked:  data/fMeans.txt
    Untracked:  data/fRegress.txt
    Untracked:  data/goGroups.txt
    Untracked:  data/mMeans.txt
    Untracked:  data/mPart.txt
    Untracked:  data/mRegress.txt
    Untracked:  data/starve-f.txt
    Untracked:  data/starve-m.txt
    Untracked:  data/xp-f.txt
    Untracked:  data/xp-m.txt
    Untracked:  figure/
    Untracked:  scoreAnalysisMulticomp.R
    Untracked:  temp.Rmd

Unstaged changes:
    Deleted:    analysis/database.Rmd
    Modified:   code/baseScript-lineComp.R
    Modified:   code/fourLinePrep.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/linReg.Rmd</code>) and HTML
(<code>docs/linReg.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/951799a1f59b0f73117a1a9fdd6090b554058cfe/analysis/linReg.Rmd" target="_blank">951799a</a>
</td>
<td>
nklimko
</td>
<td>
2023-01-08
</td>
<td>
wflow_publish("analysis/*")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nklimko/dgrp-starve/3751eb8d0ee733a7887e2cdd66a9a7ff8ef03cf7/docs/linReg.html" target="_blank">3751eb8</a>
</td>
<td>
nklimko
</td>
<td>
2022-12-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/48531146f6e7196fe50e471a9712eb58af6d11d6/analysis/linReg.Rmd" target="_blank">4853114</a>
</td>
<td>
nklimko
</td>
<td>
2022-12-14
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nklimko/dgrp-starve/e7adcbcc8bbeddfbbf080ebdc4dc745e32b64f6a/docs/linReg.html" target="_blank">e7adcbc</a>
</td>
<td>
nklimko
</td>
<td>
2022-12-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/d011087d01e7731dcfd80fda5edfa7ae3111f948/analysis/linReg.Rmd" target="_blank">d011087</a>
</td>
<td>
nklimko
</td>
<td>
2022-12-14
</td>
<td>
wflow_publish("analysis/linReg.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nklimko/dgrp-starve/e073265eccc054371c68cb9a238d06a9e7f36dd0/docs/linReg.html" target="_blank">e073265</a>
</td>
<td>
nklimko
</td>
<td>
2022-12-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/90d18bcaf0abe2a9fd0162e179f54c5b6e83d957/analysis/linReg.Rmd" target="_blank">90d18bc</a>
</td>
<td>
nklimko
</td>
<td>
2022-12-06
</td>
<td>
wflow_publish("analysis/linReg.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nklimko/dgrp-starve/03fa2812ae1580eeeca5235f6a923826553d46d0/docs/linReg.html" target="_blank">03fa281</a>
</td>
<td>
nklimko
</td>
<td>
2022-12-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/0277abe614e4a55ec3174c36c3a7918389323fb0/analysis/linReg.Rmd" target="_blank">0277abe</a>
</td>
<td>
nklimko
</td>
<td>
2022-12-06
</td>
<td>
wflow_publish("analysis/linReg.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="prep" class="section level1">
<h1>Prep</h1>
<pre class="r"><code># The following was run to prepare a data table for job submission

#expression data
xpf &lt;- fread(&quot;data/xp-f.txt&quot;)

#line/trait data
dtf &lt;- fread(&quot;data/eQTL_traits_females.csv&quot;)

#select line and starvation only, remove nulls
dtf &lt;- na.omit(dtf[,c(1,10)])

fwrite(dtf, &quot;data/starve-f.txt&quot;)

fMeans &lt;- dtf[xpf, on = .(line), nomatch=NULL, all=TRUE]

fwrite(fMeans, &quot;data/fMeans.txt&quot;)</code></pre>
<pre class="r"><code># The following was run to prepare a data table for job submission

#expression data
xpm &lt;- fread(&quot;data/xp-m.txt&quot;)

#line/trait data
dtm &lt;- fread(&quot;data/eQTL_traits_males.csv&quot;)

#select line and starvation only, remove nulls
dtm &lt;- na.omit(dtm[,c(1,11)])

fwrite(dtm, &quot;data/starve-m.txt&quot;)

mMeans &lt;- dtm[xpm, on = .(line), nomatch=NULL, all=TRUE]

fwrite(mMeans, &quot;data/mMeans.txt&quot;)</code></pre>
</div>
<div id="demo" class="section level1">
<h1>Demo</h1>
<p>As a demo, I’ve walked through the process for finding a correlation
for the first gene. Below is a plot of starvation vs flybase gene 3:</p>
<pre class="r"><code>plot(mMeans[,c(3,2), with=FALSE])</code></pre>
<p><img src="figure/linReg.Rmd/process-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-process-1">
Past versions of process-1.png
</button>
</p>
<div id="fig-process-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/e7adcbcc8bbeddfbbf080ebdc4dc745e32b64f6a/docs/figure/linReg.Rmd/process-1.png" target="_blank">e7adcbc</a>
</td>
<td>
nklimko
</td>
<td>
2022-12-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/03fa2812ae1580eeeca5235f6a923826553d46d0/docs/figure/linReg.Rmd/process-1.png" target="_blank">03fa281</a>
</td>
<td>
nklimko
</td>
<td>
2022-12-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>lm() returns the simple regression of y to x, giving parameters for a
slope-intercept form equation</p>
<pre class="r"><code>y &lt;- mMeans[,starvation]
x &lt;- mMeans[,3,with=FALSE]
lm(formula=unlist(y)~unlist(x), na.action=na.omit)</code></pre>
<pre><code>
Call:
lm(formula = unlist(y) ~ unlist(x), na.action = na.omit)

Coefficients:
(Intercept)    unlist(x)  
      22.91         5.01  </code></pre>
<p>Plotting this line shows the effect of the particular gene(x) on
starvation resistance(y).</p>
<pre class="r"><code>plot(mMeans[,c(3,2), with=FALSE])
abline(22.91, 5.01)</code></pre>
<p><img src="figure/linReg.Rmd/process-abline-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-process-abline-1">
Past versions of process-abline-1.png
</button>
</p>
<div id="fig-process-abline-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/e7adcbcc8bbeddfbbf080ebdc4dc745e32b64f6a/docs/figure/linReg.Rmd/process-abline-1.png" target="_blank">e7adcbc</a>
</td>
<td>
nklimko
</td>
<td>
2022-12-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/03fa2812ae1580eeeca5235f6a923826553d46d0/docs/figure/linReg.Rmd/process-abline-1.png" target="_blank">03fa281</a>
</td>
<td>
nklimko
</td>
<td>
2022-12-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>summary() returns a variety of useful statistics regarding simple
regression.</p>
<pre class="r"><code>summary(lm(mMeans[,c(2,3), with=FALSE], na.action=na.omit))</code></pre>
<pre><code>
Call:
lm(formula = mMeans[, c(2, 3), with = FALSE], na.action = na.omit)

Residuals:
     Min       1Q   Median       3Q      Max 
-22.6202  -6.8295  -0.4753   5.9713  24.7799 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)  
(Intercept)   22.914     11.891   1.927   0.0554 .
FBgn0000003    5.010      2.631   1.904   0.0583 .
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 9.342 on 196 degrees of freedom
Multiple R-squared:  0.01816,   Adjusted R-squared:  0.01315 
F-statistic: 3.626 on 1 and 196 DF,  p-value: 0.05835</code></pre>
<p>The p-value is what we’re interested in, and can be accessed from the
summary object at [[4]][8]</p>
<p>The following line of code combine all of these parts into one to
extract the p-value from the correlation between starvation resistance
and any gene denoted by column i.</p>
<pre class="r"><code>summary(lm(mMeans[,c(2,3), with=FALSE], na.action=na.omit))[[4]][8]</code></pre>
<p>This will need to be run over 24 thousand times to capture every gene
expressed in both females and males.</p>
</div>
<div id="correlations" class="section level1">
<h1>Correlations</h1>
<p>In order to save time, I used secretariat’s computational resources
to perform the correlation calculations in parallel. The following two
sections of code were submission scripts performed in secretariat.</p>
<pre class="r"><code>library(dplyr)
library(data.table)
library(doParallel)

registerDoParallel(cores = 12)

# file path for final table
filePath &lt;- &quot;/data/morgante_lab/nklimko/rep/dgrp-starve/data/fRegress.txt&quot;

# gene expression data from earlier bound to line and starvation
fMeans &lt;- fread(&quot;/data/morgante_lab/nklimko/rep/dgrp-starve/data/fMeans.txt&quot;)

start &lt;- 3
end &lt;- 11340
#f 11340
#m 13577

# foreach creates a list with index indicating position in loop
# function grabs the p-value from correlation of starvation to expression of trait
pvalList &lt;- foreach(i=start:end) %dopar% {  
  temp &lt;- summary(lm(fMeans[,c(2,i), with=FALSE], na.action=na.omit))[[4]][8]  
}

# converts list to vector
part &lt;- unlist(pvalList, use.names = FALSE)

# binds p-vals to column index 
id &lt;- start:end
part &lt;- data.table(id,pvalList)

# write table to file
fwrite(part, filePath)</code></pre>
<p>Both of these are available by clicking the drop down buttons on the
right -&gt;</p>
<pre class="r"><code>library(dplyr)
library(data.table)
library(doParallel)

registerDoParallel(cores = 12)

# file path for final table
filePath &lt;- &quot;/data/morgante_lab/nklimko/rep/dgrp-starve/data/mRegress.txt&quot;

# gene expression data from earlier bound to line and starvation
mMeans &lt;- fread(&quot;/data/morgante_lab/nklimko/rep/dgrp-starve/data/mMeans.txt&quot;)

start &lt;- 3
end &lt;- 13577

# foreach creates a list with index indicating position in loop
# function grabs the p-value from correlation of starvation to expression of trait
pvalList &lt;- foreach(i=start:end) %dopar% {
  
  temp &lt;- summary(lm(mMeans[,c(2,i), with=FALSE], na.action=na.omit))[[4]][8]  
}

# converts list to vector
part &lt;- unlist(pvalList, use.names = FALSE)

# binds p-vals to column index 
id &lt;- start:end
part &lt;- data.table(id,pvalList)

#writes table to file
fwrite(part, filePath)</code></pre>
<div id="qq-and-manthattan-data-prep--" class="section level4">
<h4>QQ and Manthattan Data Prep -&gt;</h4>
<pre class="r"><code># Female qq prep

#f 11340
#m 13577

gg &lt;- vector(mode=&#39;list&#39;, length=4)

#read in results from compute node
fReg &lt;- fread(&quot;data/fRegress.txt&quot;)

#number of data points
n &lt;- dim(fReg)[1]

#Theoretical Quantiles
uniform &lt;- 1:n/(n+1)

#sorted p values
empirical &lt;- sort(fReg[,pvalList])

#-log 10
logPlot &lt;- -log10(fReg[,pvalList])

#table for qqplot
qqdata &lt;- data.table(uniform, empirical, logPlot)

qqdataF &lt;- qqdata

# ggplot of data to fpr, qqplot
if(FALSE){
gg[[1]] &lt;- ggplot(qqdata, aes(x=uniform, y=empirical)) +
  geom_point(color=&quot;red&quot;) +  
  geom_segment(aes(x=0,y=0,xend=1,yend=1)) + 
  xlab(&quot;Theoretical Quantiles&quot;) +
  ylab(&quot;Sorted p-values&quot;) +
  ggtitle(&quot;Female Expression QQ Plot&quot;)
}

gg[[2]] &lt;- ggplot(qqdata, aes(x=uniform, y=logPlot)) +
  geom_point(color=&quot;red&quot;) + 
  geom_hline(yintercept= -log10(0.05/n), color=&quot;magenta&quot;) +
  xlab(&quot;Theoretical Quantiles&quot;) +
  ylab(&quot;-log10 p-values&quot;) +
  ggtitle(&quot;Female Expression Manhattan Plot&quot;)



## Code for looking at specific genes
#fMeans &lt;- fread(&quot;data/fMeans.txt&quot;)
#genes &lt;- colnames(fMeans)
#genes &lt;- genes[3:length(genes)]
#fReg &lt;- fReg[, gene:=genes]</code></pre>
<pre class="r"><code># Male qq prep

#read in results from compute node
mReg &lt;- fread(&quot;data/mRegress.txt&quot;)

#number of data points
n &lt;- dim(mReg)[1]

#Theoretical Quantiles
uniform &lt;- 1:n/(n+1)

#sorted p values
empirical &lt;- sort(mReg[,pvalList])

#-log 10
logPlot &lt;- -log10(mReg[,pvalList])

#table for qqplot
qqdata &lt;- data.table(uniform, empirical, logPlot)

qqdataM &lt;- qqdata


# ggplot of data to fpr, qqplot
if(FALSE){
gg[[3]] &lt;- ggplot(qqdata, aes(x=uniform, y=empirical)) +
  geom_point(color=&quot;blue&quot;) +  
  geom_segment(aes(x=0,y=0,xend=1,yend=1), linewidth=1) + 
  xlab(&quot;Theoretical Quantiles&quot;) +
  ylab(&quot;Sorted p-values&quot;) +
  ggtitle(&quot;Male Expression QQ Plot&quot;)
}


gg[[4]] &lt;- ggplot(qqdata, aes(x=uniform, y=logPlot)) +
  geom_point(color=&quot;blue&quot;) + 
  geom_hline(yintercept= -log10(0.05/n), color=&quot;cyan&quot;) +
  xlab(&quot;Theoretical Quantiles&quot;) +
  ylab(&quot;-log10 p-values&quot;) +
  ggtitle(&quot;Male Expression Manhattan Plot&quot;)

## Code for looking at specific genes
#mMeans &lt;- fread(&quot;data/mMeans.txt&quot;)
#genes &lt;- colnames(mMeans)
#genes &lt;- genes[3:length(genes)]
#mReg &lt;- mReg[, gene:=genes]</code></pre>
</div>
</div>
<div id="qq-plots" class="section level1">
<h1>QQ Plots</h1>
<pre class="r"><code>par(mfrow=c(1,2))

qq(qqdataF[,empirical], main=&quot;Female Gene p-values&quot;)
qq(qqdataM[,empirical], main=&quot;Male Gene p-values&quot;)</code></pre>
<p><img src="figure/linReg.Rmd/qq-plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-qq-plot-1">
Past versions of qq-plot-1.png
</button>
</p>
<div id="fig-qq-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/e7adcbcc8bbeddfbbf080ebdc4dc745e32b64f6a/docs/figure/linReg.Rmd/qq-plot-1.png" target="_blank">e7adcbc</a>
</td>
<td>
nklimko
</td>
<td>
2022-12-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="manhattan-plots" class="section level1">
<h1>Manhattan Plots</h1>
<pre class="r"><code>par(mfrow=c(1,2))

plot_grid(gg[[2]],gg[[4]], ncol=2)</code></pre>
<p><img src="figure/linReg.Rmd/manhattan-plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-manhattan-plot-1">
Past versions of manhattan-plot-1.png
</button>
</p>
<div id="fig-manhattan-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/e7adcbcc8bbeddfbbf080ebdc4dc745e32b64f6a/docs/figure/linReg.Rmd/manhattan-plot-1.png" target="_blank">e7adcbc</a>
</td>
<td>
nklimko
</td>
<td>
2022-12-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Part of the reason both plots deviate from linearity because we did
not account for genes that were correlated with each other.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /opt/ohpc/pub/Software/openblas_0.3.10/lib/libopenblas_haswellp-r0.3.10.dev.so

locale:
 [1] LC_CTYPE=en_US.utf-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.utf-8        LC_COLLATE=en_US.utf-8    
 [5] LC_MONETARY=en_US.utf-8    LC_MESSAGES=en_US.utf-8   
 [7] LC_PAPER=en_US.utf-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.utf-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] qqman_0.1.8       cowplot_1.1.1     ggplot2_3.3.5     data.table_1.14.2
[5] dplyr_1.0.8       workflowr_1.7.0  

loaded via a namespace (and not attached):
 [1] tidyselect_1.1.2 xfun_0.30        bslib_0.3.1      purrr_0.3.4     
 [5] colorspace_2.0-3 vctrs_0.4.1      generics_0.1.2   htmltools_0.5.2 
 [9] yaml_2.3.5       utf8_1.2.2       rlang_1.0.4      later_1.3.0     
[13] pillar_1.7.0     jquerylib_0.1.4  withr_2.5.0      glue_1.6.2      
[17] DBI_1.1.2        calibrate_1.7.7  lifecycle_1.0.1  stringr_1.4.0   
[21] munsell_0.5.0    gtable_0.3.0     evaluate_0.15    labeling_0.4.2  
[25] knitr_1.38       callr_3.7.0      fastmap_1.1.0    httpuv_1.6.5    
[29] ps_1.6.0         fansi_1.0.3      highr_0.9        Rcpp_1.0.8.3    
[33] promises_1.2.0.1 scales_1.2.0     jsonlite_1.8.0   farver_2.1.0    
[37] fs_1.5.2         digest_0.6.29    stringi_1.7.6    processx_3.5.3  
[41] getPass_0.2-2    rprojroot_2.0.3  grid_4.0.3       cli_3.3.0       
[45] tools_4.0.3      magrittr_2.0.3   sass_0.4.1       tibble_3.1.6    
[49] crayon_1.5.1     whisker_0.4      pkgconfig_2.0.3  MASS_7.3-56     
[53] ellipsis_0.3.2   assertthat_0.2.1 rmarkdown_2.16   httr_1.4.2      
[57] rstudioapi_0.13  R6_2.5.1         git2r_0.30.1     compiler_4.0.3  </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
