<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Iterative Comparison - Male</title>

<script src="site_libs/header-attrs-2.20/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">dgrp-starve</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/nklimko/dgrp-starve">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Iterative Comparison - Male</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-03-13
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>dgrp-starve/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221101code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221101)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221101code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221101)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomnklimkodgrpstarvetree97bd277c1dd36e5b7dbe179f5aef1b657432aabetargetblank97bd277a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/nklimko/dgrp-starve/tree/97bd277c1dd36e5b7dbe179f5aef1b657432aabe" target="_blank">97bd277</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomnklimkodgrpstarvetree97bd277c1dd36e5b7dbe179f5aef1b657432aabetargetblank97bd277a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/nklimko/dgrp-starve/tree/97bd277c1dd36e5b7dbe179f5aef1b657432aabe" target="_blank">97bd277</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData

Untracked files:
    Untracked:  analysis/linearReg.Rmd
    Untracked:  code/PCA/
    Untracked:  code/data-prep/
    Untracked:  code/fabio/
    Untracked:  code/intro-starve/
    Untracked:  code/methodComp/
    Untracked:  code/regress/
    Untracked:  data/corLoop-f.rds
    Untracked:  data/corLoop-m.rds
    Untracked:  data/eQTL_traits_females.csv
    Untracked:  data/eQTL_traits_males.csv
    Untracked:  data/fRegress.txt
    Untracked:  data/fRegress_adj.txt
    Untracked:  data/gbayesC-f.Rds
    Untracked:  data/gbayesC-m.Rds
    Untracked:  data/gbayesC.Rds
    Untracked:  data/gbayes_100k-f.Rds
    Untracked:  data/gbayes_100k-m.Rds
    Untracked:  data/goGroups.txt
    Untracked:  data/mPart.txt
    Untracked:  data/mRegress.txt
    Untracked:  data/mRegress_adj.txt
    Untracked:  data/multiReg.rData
    Untracked:  data/starve-f.txt
    Untracked:  data/starve-m.txt
    Untracked:  data/xp-f.txt
    Untracked:  data/xp-m.txt
    Untracked:  data/y_save.txt
    Untracked:  figure/
    Untracked:  git2r_0.31.0.tar.gz
    Untracked:  notes/
    Untracked:  workflowr_1.7.0.tar.gz

Unstaged changes:
    Deleted:    analysis/gremlo.R
    Deleted:    analysis/stepwise-f.Rmd
    Deleted:    analysis/stepwise-m.Rmd
    Deleted:    analysis/testing.R
    Modified:   analysis/trace.Rmd
    Deleted:    code/baseScript-lineComp.R
    Deleted:    code/combineSNP.R
    Deleted:    code/four-comp.76979.err
    Deleted:    code/four-comp.76979.out
    Deleted:    code/four-comp.sbatch
    Deleted:    code/fourLinePrep.R
    Deleted:    code/line_avgMinus.R
    Deleted:    code/line_avgPlus.R
    Deleted:    code/line_difMinus.R
    Deleted:    code/line_difPlus.R
    Deleted:    code/snpGene.R
    Deleted:    code/starveDataPrep.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/methodComp-m.Rmd</code>) and HTML
(<code>docs/methodComp-m.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/97bd277c1dd36e5b7dbe179f5aef1b657432aabe/analysis/methodComp-m.Rmd" target="_blank">97bd277</a>
</td>
<td>
nklimko
</td>
<td>
2023-03-13
</td>
<td>
wflow_publish("analysis/methodComp-*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nklimko/dgrp-starve/d924e6a5fddb3db2f5b401f51730b4d2db15efdb/docs/methodComp-m.html" target="_blank">d924e6a</a>
</td>
<td>
nklimko
</td>
<td>
2023-03-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/e7abfc8eddf1d480369700f38f23c22a19817565/analysis/methodComp-m.Rmd" target="_blank">e7abfc8</a>
</td>
<td>
nklimko
</td>
<td>
2023-03-06
</td>
<td>
wflow_publish("analysis/methodComp-*")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nklimko/dgrp-starve/5fc994a9124570a09207440a8a419fcd8f5b23ae/docs/methodComp-m.html" target="_blank">5fc994a</a>
</td>
<td>
nklimko
</td>
<td>
2023-03-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/05bbee26c878f8b38c0af8ee66d11d0a1c851306/analysis/methodComp-m.Rmd" target="_blank">05bbee2</a>
</td>
<td>
nklimko
</td>
<td>
2023-03-05
</td>
<td>
wflow_publish("analysis/methodComp-m.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>The overarching goal of this process is to predict the phenotype for
starvation resistance, a continuous trait, by using gene expression
data, another continuous trait. This is done using k-fold cross
validation to create models based on a subset of the data and
calculating the correlation of that model with the remaining partition.
By repeating this process multiple times with different training and
testing partitions, model bias can be significantly reduced and allows
for calculation of average correlation coefficients for each model. The
primary difference between the methods in question is the prior
distribution used.</p>
<pre class="r"><code>#loop count and data limit
iter &lt;- 48

#Parallel core count
#registerDoParallel(cores = 8)

#ggplot holder list
gg &lt;- vector(mode=&#39;list&#39;, length=12)

# result storage elements
fit_greml &lt;- vector(mode=&#39;list&#39;, length=iter)
fit_gbayesC &lt;- vector(mode=&#39;list&#39;, length=iter)
fit_varbvs &lt;- vector(mode=&#39;list&#39;, length=iter)
fit_glmnet &lt;- vector(mode=&#39;list&#39;, length=iter)</code></pre>
<p>The first main set of data used for this analysis is a matrix of gene
expression by DGRP line matched to raw starvation resistance. A second
cluster of data sets provided by the Morgante Lab includes information
on Wolbachia infection status and inversion status by line along with
functions to adjust phenotypic values based on these two factors.</p>
<pre class="r"><code>dataFlag &lt;- TRUE

if(dataFlag){

 #wolb infection and inversion status data with phenotype adjustment function
load(&quot;/data2/morgante_lab/data/dgrp/misc/adjustData.RData&quot;)

  
  #expression data matched to line and starvation phenotype
  #xp_f &lt;- fread(&quot;data/xp-f.txt&quot;)
  
  xp_m &lt;- fread(&quot;data/xp-m.txt&quot;)
  
  #setwd(&quot;C:/Users/noahk/OneDrive/Desktop/amogus&quot;)
  #getwd()


  #create matrix of only gene expression, trims line and starvation
  X &lt;- as.matrix(xp_m[,3:13577])
  rownames(X) &lt;- xp_m[,line]
  W &lt;- scale(X)
  
  y_temp &lt;- xp_m[,starvation]
  dat &lt;- data.frame(id=xp_m[,line], y=y_temp)
  y &lt;- adjustPheno(dat, &quot;starvation&quot;)

} else{
  
  # Toy Data set, 200x100 matrix
  W &lt;- matrix(rnorm(20000), ncol = 100)
  colnames(W) &lt;- paste0(&quot;gene&quot;, 1:ncol(W))
  rownames(W) &lt;- paste0(&quot;line&quot;, 1:nrow(W))
  
  #model uses genes 1:5 and 10:20
  y &lt;- rowSums(W[, 1:5]) + rowSums(W[, 10:20]) + rnorm(nrow(W))

}</code></pre>
<pre><code>Type III ANOVA table for covariates: starvation
                 Df Sum of Sq   RSS    AIC F value Pr(&gt;F)
&lt;none&gt;                        15975 893.32               
factor(wolba)     1     3.041 15978 891.35  0.0354 0.8509
factor(In_2L_t)   2   299.141 16274 892.99  1.7415 0.1781
factor(In_2R_NS)  2   243.044 16218 892.31  1.4149 0.2455
factor(In_3R_P)   2   230.105 16205 892.15  1.3396 0.2645
factor(In_3R_K)   2   288.050 16263 892.85  1.6770 0.1898
factor(In_3R_Mo)  2   207.445 16182 891.87  1.2077 0.3012


Estimated effects
                    Estimate Std. Error    t value     Pr(&gt;|t|)
(Intercept)       45.6614374   1.157976 39.4321227 2.909220e-92
factor(wolba)y    -0.2574606   1.368248 -0.1881681 8.509500e-01
factor(In_2L_t)1   1.7198329   2.311372  0.7440744 4.577704e-01
factor(In_2L_t)2  -3.7780449   2.348592 -1.6086422 1.093906e-01
factor(In_2R_NS)1 -1.0811449   3.443957 -0.3139252 7.539297e-01
factor(In_2R_NS)2  5.8594026   3.595753  1.6295343 1.048923e-01
factor(In_3R_P)1  -2.0731924   3.879580 -0.5343858 5.937128e-01
factor(In_3R_P)2   7.1994131   4.718118  1.5259077 1.287315e-01
factor(In_3R_K)1   5.3513957   3.043456  1.7583284 8.033635e-02
factor(In_3R_K)2   3.7673764   6.668091  0.5649857 5.727643e-01
factor(In_3R_Mo)1 -3.0569154   3.189101 -0.9585510 3.390293e-01
factor(In_3R_Mo)2 -3.0977753   2.402719 -1.2892793 1.989020e-01</code></pre>
<p>The matrix containing only gene expression by line data was then
scaled to an absolute max of 1. along with this, a Translation
Relationship Matrix was generated by taking the crossproduct of the
scaled expression matrix and scaling it down by the number of genes.</p>
<pre class="r"><code>#model to solve for, vector of ones
mu &lt;- matrix(rep(1, length(y)), ncol=1)
#names(mu) &lt;- paste0(&quot;line&quot;, 1:length(mu))
rownames(mu) &lt;- xp_m[,line]
TRM &lt;- tcrossprod(W)/ncol(W)

# k-fold parameters
n &lt;- length(y)
fold &lt;- 10
iter &lt;- 48</code></pre>
<p>10 was chosen for k-fold cross validation resulting in 19 lines per
validation set and 179 lines per training set.</p>
<pre class="r"><code>### sample analysis of gbayesC to show that convergence is working as expected 

test_IDs &lt;- sample(1:n, as.integer(n / fold))
  
  W_train &lt;- W[-test_IDs,]
  W_test &lt;- W[test_IDs,]
  y_train &lt;- y[-test_IDs]
  y_test &lt;- y[test_IDs]
  

  ### GBAYES-C
  
  
  fitC &lt;- qgg::gbayes(y=y_train, W=W_train, method=&quot;bayesC&quot;, scaleY=FALSE, nit=10000, nburn=5000)</code></pre>
<pre class="r"><code>plotCustomBayes &lt;- function(rdsPath){

# data read in is a gBayes fit object from qgg
#fitC &lt;- readRDS(&quot;gbayesC-f.Rds&quot;)
fitC &lt;- readRDS(rdsPath)

# calculate column narrow heritability
fitC$h2 &lt;- fitC$vgs/(fitC$vgs+fitC$ves)

# data is extracted and stored
fitData &lt;- data.table(iter=1:10000, ves=fitC$ves, vbs=fitC$vbs, vgs=fitC$vgs, h2=fitC$h2)

gg &lt;- vector(mode=&#39;list&#39;, length=4)

gg[[1]] &lt;- ggplot(fitData, aes(x=iter, y=ves)) +
  geom_point(size=0.5) + 
  labs(x=&quot;Iteration&quot;, y=&quot;Ve&quot;) + 
  ggtitle(&quot;Posterior Mean for Residual Variance&quot;)

gg[[2]] &lt;- ggplot(fitData, aes(x=iter, y=vbs)) +
  geom_point(size=0.5) + 
  labs(x=&quot;Iteration&quot;, y=&quot;Vb&quot;) + 
  ggtitle(&quot;Posterior Mean for Marker Variance&quot;)

gg[[3]] &lt;- ggplot(fitData, aes(x=iter, y=vgs)) +
  geom_point(size=0.5) + 
  labs(x=&quot;Iteration&quot;, y=&quot;Vg&quot;) + 
  ggtitle(&quot;Posterior Mean for Genomic Variance&quot;)

gg[[4]] &lt;- ggplot(fitData, aes(x=iter, y=ves)) +
  geom_point(size=0.5) + 
  labs(x=&quot;Iteration&quot;, y=&quot;h^2&quot;) + 
  ggtitle(&quot;Posterior Mean for Narrow-sense Heritability&quot;)

  return(gg) 
}

    # fit taken from one iteration, restored from Rds using code chunk shown above
    #fitC &lt;- readRDS(&quot;data/gbayesC-m.Rds&quot;)
    
    plotHold &lt;- plotCustomBayes(&quot;data/gbayesC-m.Rds&quot;)

    plot_grid(plotHold[[1]],plotHold[[2]], ncol=2)
    plot_grid(plotHold[[3]],plotHold[[4]], ncol=2)
    #plotBayes(fit=fitC, what=&quot;trace&quot;)</code></pre>
<p>The following methods are currently implemented:</p>
<p>qgg::greml - Genomic Restricted Maximum Likelihood Estimation using
Best Linear Unbiased Predictor. GREML uses a Gaussian prior
distribution, performing the least shrinkage and no variable
selection.</p>
<p>glmnet - glmnet uses LASSO, or least absolute shrinkage and selection
operator. Bayesian LASSO uses a thick-tailed prior which performs
greater shrinkage towards the mean than a Gaussian distribution.</p>
<p>qgg::gbayes - BayesC has been implemented using this command. BayesC
uses a spike-slab prior which performs variable selection. This
intentionally sets the effect of some genes expressed to zero which
models the idea that some genes have no effect on the given trait.</p>
<p>varbvs - Bayesian variable selection performs variable selection
using another spike-slab prior. This method avoids Markov Chain Monte
Carlo methods by approximating the posterior distribution to reduce
computational resources.</p>
<pre class="r"><code>#this runs on the same data sets as greml

### qgg_gBayesC
### https://www.rdocumentation.org/packages/qgg/versions/1.1.1/topics/gbayes

# Bayes C: uses a “rounded spike” (low-variance Gaussian) at origin many small effects can contribute to polygenic component, reduces the dimensionality of the model (makes Gibbs sampling feasible).

#Parallel Header
#tempResult &lt;- 

iter &lt;- 1
corLoop &lt;- foreach(i=1:iter) %dopar% {

#Linear Header
#for(i in 1:iter){
  
  corResult &lt;- (1:4)
  
  
  
  #setup train and test sets with trait vectors
  test_IDs &lt;- sample(1:n, as.integer(n / fold))
  
  W_train &lt;- W[-test_IDs,]
  W_test &lt;- W[test_IDs,]
  y_train &lt;- y[-test_IDs]
  y_test &lt;- y[test_IDs]
  
  
  ### GREML, qgg package
  fitGreml &lt;- qgg::greml(y=y, X=mu, GRM=list(A=TRM), validate = matrix(test_IDs,ncol=1), verbose=FALSE)
  
  #Store coeff directly
  fit_greml[[i]] &lt;- fitGreml$accuracy$Corr
  
  corResult[1] &lt;- fitGreml$accuracy$Corr

  
  
  ### GBAYES-C
  
  
  fitC &lt;- qgg::gbayes(y=y_train, W=W_train, method=&quot;bayesC&quot;, scaleY=FALSE, nit=10000, nburn=5000)
  
  # expected/calculated value for y_test
  # \hat{y}_test = W_{test} * \hat{b} + \hat{mu}
  y_calc &lt;- W_test %*% fitC$b + mean(y_train)
  
  # store coeff
  fit_gbayesC[[i]] &lt;- cor(y_test, y_calc)
  
  corResult[2] &lt;- cor(y_test, y_calc)
  
  if(i==1){
    print(paste0(&quot;gbayesC trace plots from validate set &quot;,i))
    plotBayes(fit=list(fitC), what = &quot;trace&quot;)
  }
    
  
  ### VARBVS
  fitVarb &lt;- varbvs::varbvs(X = W_train, NULL, y=y_train, family = &quot;gaussian&quot;, logodds=seq(-3.5,-1,0.1), sa = 1, verbose=FALSE)
  
  # \hat{y}_test = W_{test} * \hat{b} + \hat{mu}
  y_calc &lt;- W_test %*% fitVarb$beta + mean(y_train)
  
  fit_varbvs[[i]] &lt;- cor(y_test, y_calc)
  
  corResult[3] &lt;- cor(y_test, y_calc)
  
  
  
  
  ### GLMNET  
  fitlm &lt;- glmnet::cv.glmnet(x=W_train, y=y_train, alpha=1)
  

  b_hat &lt;- glmnet::coef.glmnet(fitlm, s=&quot;lambda.min&quot;)


  y_int &lt;- b_hat[1]

  b_hat &lt;- b_hat[2:length(b_hat)]

  y_calc &lt;- W_test %*% b_hat + y_int

  fit_glmnet[[i]] &lt;- cor(y_test, y_calc)
  
  corResult[4] &lt;- cor(y_test, y_calc)
  
  
  corResult

}</code></pre>
<pre class="r"><code># results loaded from correlation loop structure
corLoop &lt;- readRDS(&quot;data/corLoop-m.rds&quot;)



for(i in 1:iter){
  fit_greml[[i]] &lt;- corLoop[[i]][1]
  fit_gbayesC[[i]] &lt;- corLoop[[i]][2]
  fit_varbvs[[i]] &lt;- corLoop[[i]][3]
  fit_glmnet[[i]] &lt;- corLoop[[i]][4]
}</code></pre>
<pre class="r"><code>iter &lt;- 48

bint &lt;- 10

### qgg_greml

qgg_greml_data &lt;- as.data.table(fit_greml)
qgg_greml_data &lt;- transpose(qgg_greml_data)
colnames(qgg_greml_data) &lt;- &quot;cor&quot;

#print(paste0(&quot;GREML&quot;))
#print(paste0(&quot;Mean correlation coefficient: &quot;, mean(qgg_greml_data[,cor])))
#print(paste0(&quot;Variance of correlation coefficient: &quot;, var(qgg_greml_data)))

gg[[1]] &lt;- ggplot(qgg_greml_data, aes(x=cor)) +
  geom_histogram(bins=bint, fill=&#39;red&#39;) +
  labs(x=&quot;Corr Coeff&quot;) +
  ggtitle(&quot;GREML CV Correlations Histogram&quot;)


### qgg_gbayesC

qgg_gbayesC_data &lt;- as.data.table(fit_gbayesC)
qgg_gbayesC_data &lt;- transpose(qgg_gbayesC_data)
colnames(qgg_gbayesC_data) &lt;- &quot;cor&quot;

#print(paste0(&quot;gBayesC&quot;))
#print(paste0(&quot;Mean correlation coefficient: &quot;, mean(qgg_gbayesC_data[,cor])))
#print(paste0(&quot;Variance of correlation coefficient: &quot;, var(qgg_gbayesC_data)))

gg[[2]] &lt;- ggplot(qgg_gbayesC_data, aes(x=cor)) +
  geom_histogram(bins=bint, fill=&#39;red&#39;) +
  labs(x=&quot;Corr Coeff&quot;) +
  ggtitle(&quot;gBayesC Prediction Corr Histogram&quot;)


### VARBVS

varbvs_data &lt;- as.data.table(fit_varbvs)
varbvs_data &lt;- transpose(varbvs_data)
colnames(varbvs_data) &lt;- &quot;cor&quot;

#print(paste0(&quot;VARBVS&quot;))
#print(paste0(&quot;Mean correlation coefficient: &quot;, mean(varbvs_data[,cor])))
#print(paste0(&quot;Variance of correlation coefficient: &quot;, var(varbvs_data)))

gg[[3]] &lt;- ggplot(varbvs_data, aes(x=cor)) +
  geom_histogram(bins=bint, fill=&#39;red&#39;) +
  labs(x=&quot;Corr Coeff&quot;) +
  ggtitle(&quot;varbvs Prediction Corr Histogram&quot;)


### GLMNET

glmnet_data &lt;- as.data.table(fit_glmnet)
glmnet_data &lt;- transpose(glmnet_data)
colnames(glmnet_data) &lt;- &quot;cor&quot;

#print(paste0(&quot;GLMNET&quot;))
#print(paste0(&quot;Mean correlation coefficient: &quot;, mean(glmnet_data[,cor])))
#print(paste0(&quot;Variance of correlation coefficient: &quot;, var(glmnet_data)))

gg[[4]] &lt;- ggplot(glmnet_data, aes(x=cor)) +
  geom_histogram(bins=bint, fill=&#39;red&#39;) +
  labs(x=&quot;Corr Coeff&quot;) +
  ggtitle(&quot;glmnet Prediction Corr Histogram&quot;)</code></pre>
<pre class="r"><code>result &lt;- data.table(method=&quot;greml&quot;, meanCoeff=mean(qgg_greml_data[,cor])) 
result[,varCoeff := var(qgg_greml_data)]
                    
temp &lt;- data.table(method=&quot;gBayesC&quot;, meanCoeff=mean(qgg_gbayesC_data[,cor]))
temp[,varCoeff := var(qgg_gbayesC_data)]
result &lt;- rbind(result, temp)

temp &lt;- data.table(method=&quot;varbvs&quot;, meanCoeff=mean(varbvs_data[,cor]))
temp[,varCoeff := var(varbvs_data)]
result &lt;- rbind(result, temp)

temp &lt;- data.table(method=&quot;glmnet&quot;, meanCoeff=mean(glmnet_data[,cor]))
temp[,varCoeff := var(glmnet_data)]
result &lt;- rbind(result, temp)

# Map the time of day to different fill colors
#gg[[5]] &lt;- ggplot(data=result, aes(x=method, y=meanCoeff, fill=method)) +
    #geom_bar(stat=&quot;identity&quot;)

#boxplot(result$meanCoeff)

gg[[5]] &lt;- ggplot(data=result, aes(x=method, y=meanCoeff, fill=method)) +
    geom_bar(stat=&quot;identity&quot;) +
    ggtitle(&quot;Mean Correlation Coefficient by Method&quot;)

gg[[6]] &lt;- ggplot(data=result, aes(x=method, y=varCoeff, fill=method)) +
    geom_bar(stat=&quot;identity&quot;) +
    ggtitle(&quot;Variance of Coefficients by Method&quot;)



#gg[[5]] &lt;- ggplot(result, aes(x=meanCoeff)) + 
#  geom_boxplot() + 
#  coord_flip()

#gg[[6]] &lt;- ggplot(result, aes(x=varCoeff)) + 
#  geom_boxplot() + 
#  coord_flip()</code></pre>
<div id="correlation-coefficient-histograms" class="section level3">
<h3>Correlation Coefficient Histograms</h3>
<pre class="r"><code>plot_grid(gg[[1]],gg[[2]],gg[[3]],gg[[4]], ncol=2)</code></pre>
<p><img src="figure/methodComp-m.Rmd/plot-print-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="method-comparison-summary" class="section level3">
<h3>Method Comparison Summary</h3>
<pre class="r"><code>plot_grid(gg[[5]],gg[[6]], ncol=2)</code></pre>
<p><img src="figure/methodComp-m.Rmd/plot-print2-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Rocky Linux 8.5 (Green Obsidian)

Matrix products: default
BLAS/LAPACK: /opt/ohpc/pub/libs/gnu9/openblas/0.3.7/lib/libopenblasp-r0.3.7.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] glmnet_4.1-6      Matrix_1.5-3      varbvs_2.5-16     qgg_1.1.1        
 [5] doParallel_1.0.17 iterators_1.0.14  foreach_1.5.2     qqman_0.1.8      
 [9] cowplot_1.1.1     ggplot2_3.4.1     data.table_1.14.8 dplyr_1.1.0      
[13] workflowr_1.7.0  

loaded via a namespace (and not attached):
 [1] httr_1.4.5          sass_0.4.5          nor1mix_1.3-0      
 [4] jsonlite_1.8.4      splines_4.1.2       bslib_0.4.2        
 [7] getPass_0.2-2       statmod_1.5.0       highr_0.10         
[10] latticeExtra_0.6-30 yaml_2.3.7          pillar_1.8.1       
[13] lattice_0.20-45     quantreg_5.94       glue_1.6.2         
[16] digest_0.6.31       RColorBrewer_1.1-3  promises_1.2.0.1   
[19] colorspace_2.1-0    htmltools_0.5.4     httpuv_1.6.9       
[22] pkgconfig_2.0.3     SparseM_1.81        calibrate_1.7.7    
[25] scales_1.2.1        processx_3.8.0      whisker_0.4.1      
[28] jpeg_0.1-10         later_1.3.0         MatrixModels_0.5-1 
[31] git2r_0.31.0        tibble_3.1.8        generics_0.1.3     
[34] farver_2.1.1        cachem_1.0.7        withr_2.5.0        
[37] cli_3.6.0           survival_3.5-3      magrittr_2.0.3     
[40] deldir_1.0-6        mcmc_0.9-7          evaluate_0.20      
[43] ps_1.7.2            fs_1.6.1            fansi_1.0.4        
[46] MASS_7.3-58.2       tools_4.1.2         lifecycle_1.0.3    
[49] stringr_1.5.0       MCMCpack_1.6-3      interp_1.1-3       
[52] munsell_0.5.0       callr_3.7.3         compiler_4.1.2     
[55] jquerylib_0.1.4     rlang_1.0.6         grid_4.1.2         
[58] rstudioapi_0.14     labeling_0.4.2      rmarkdown_2.20     
[61] gtable_0.3.1        codetools_0.2-19    R6_2.5.1           
[64] knitr_1.42          fastmap_1.1.1       utf8_1.2.3         
[67] rprojroot_2.0.3     shape_1.4.6         stringi_1.7.12     
[70] Rcpp_1.0.10         vctrs_0.5.2         png_0.1-8          
[73] tidyselect_1.2.0    xfun_0.37           coda_0.19-4        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
