<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Script Progression</title>

<script src="site_libs/header-attrs-2.16/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">dgrp-starve</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/nklimko/dgrp-starve">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Script Progression</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-01-10
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>dgrp-starve/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221101code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221101)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221101code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221101)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/data/morgante_lab/nklimko/rep/dgrp-starve/data/starve.csv
</td>
<td style="text-align:left;">
data/starve.csv
</td>
</tr>
<tr>
<td style="text-align:left;">
/data/morgante_lab/nklimko/rep/dgrp-starve/data/INPUT.txt
</td>
<td style="text-align:left;">
data/INPUT.txt
</td>
</tr>
<tr>
<td style="text-align:left;">
/data/morgante_lab/nklimko/rep/dgrp-starve/data/CHANGE.txt
</td>
<td style="text-align:left;">
data/CHANGE.txt
</td>
</tr>
<tr>
<td style="text-align:left;">
/data/morgante_lab/nklimko/rep/dgrp-starve/data/snpList.txt
</td>
<td style="text-align:left;">
data/snpList.txt
</td>
</tr>
<tr>
<td style="text-align:left;">
/data/morgante_lab/nklimko/rep/dgrp-starve/data/geneHits.txt
</td>
<td style="text-align:left;">
data/geneHits.txt
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomnklimkodgrpstarvetree5cc6e1887b293b3f1e79ce2e57c71c8b6bb3812ctargetblank5cc6e18a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/nklimko/dgrp-starve/tree/5cc6e1887b293b3f1e79ce2e57c71c8b6bb3812c" target="_blank">5cc6e18</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomnklimkodgrpstarvetree5cc6e1887b293b3f1e79ce2e57c71c8b6bb3812ctargetblank5cc6e18a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/nklimko/dgrp-starve/tree/5cc6e1887b293b3f1e79ce2e57c71c8b6bb3812c" target="_blank">5cc6e18</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  code/aaaTest
    Untracked:  code/analysisSR.R
    Untracked:  code/geneGO.R
    Untracked:  code/multiPrep.R
    Untracked:  code/regress.81916.err
    Untracked:  code/regress.81916.out
    Untracked:  code/regress.81918.err
    Untracked:  code/regress.81918.out
    Untracked:  code/regress.R
    Untracked:  code/regress.sbatch
    Untracked:  code/regressF.81919.err
    Untracked:  code/regressF.81919.out
    Untracked:  code/regressF.R
    Untracked:  code/regressF.sbatch
    Untracked:  code/snpGene.77509.err
    Untracked:  code/snpGene.77509.out
    Untracked:  code/snpGene.77515.err
    Untracked:  code/snpGene.77515.out
    Untracked:  code/snpGene.sbatch
    Untracked:  data/eQTL_traits_females.csv
    Untracked:  data/eQTL_traits_males.csv
    Untracked:  data/fMeans.txt
    Untracked:  data/fRegress.txt
    Untracked:  data/goGroups.txt
    Untracked:  data/mMeans.txt
    Untracked:  data/mPart.txt
    Untracked:  data/mRegress.txt
    Untracked:  data/starve-f.txt
    Untracked:  data/starve-m.txt
    Untracked:  data/xp-f.txt
    Untracked:  data/xp-m.txt
    Untracked:  figure/
    Untracked:  lmm.R
    Untracked:  qqdum.R
    Untracked:  scoreAnalysisMulticomp.R
    Untracked:  temp.Rmd

Unstaged changes:
    Deleted:    analysis/database.Rmd
    Modified:   code/baseScript-lineComp.R
    Modified:   code/fourLinePrep.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/scripts.Rmd</code>) and HTML
(<code>docs/scripts.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nklimko/dgrp-starve/9e86f566dcc9971c872790fc30a35d6c87be660f/docs/scripts.html" target="_blank">9e86f56</a>
</td>
<td>
nklimko
</td>
<td>
2023-01-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nklimko/dgrp-starve/3751eb8d0ee733a7887e2cdd66a9a7ff8ef03cf7/docs/scripts.html" target="_blank">3751eb8</a>
</td>
<td>
nklimko
</td>
<td>
2022-12-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nklimko/dgrp-starve/88ee9ec7d3879950d1c8eb4630670e808ea4dcc6/docs/scripts.html" target="_blank">88ee9ec</a>
</td>
<td>
nklimko
</td>
<td>
2022-11-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nklimko/dgrp-starve/37278c96b8526dfc781a9c5b2f075fac1869b045/docs/scripts.html" target="_blank">37278c9</a>
</td>
<td>
nklimko
</td>
<td>
2022-11-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nklimko/dgrp-starve/f09779ec190667ec4bd3e7d1b1dd24f717eb9970/docs/scripts.html" target="_blank">f09779e</a>
</td>
<td>
nklimko
</td>
<td>
2022-11-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nklimko/dgrp-starve/f1d6437766ad9ecd416ac54f034cbad7f7bdcf9d/docs/scripts.html" target="_blank">f1d6437</a>
</td>
<td>
nklimko
</td>
<td>
2022-11-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nklimko/dgrp-starve/79e98ca129310232141bbab69da3f597907100ae/docs/scripts.html" target="_blank">79e98ca</a>
</td>
<td>
nklimko
</td>
<td>
2022-11-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nklimko/dgrp-starve/d23d7f71489e3ed50d081bccd668c65a499071cc/docs/scripts.html" target="_blank">d23d7f7</a>
</td>
<td>
nklimko
</td>
<td>
2022-11-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/3a282f1db3d21899f3e0d61f62defef61b7de01c/analysis/scripts.Rmd" target="_blank">3a282f1</a>
</td>
<td>
nklimko
</td>
<td>
2022-11-11
</td>
<td>
wflow_publish("analysis/*")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="sr-data-prep" class="section level2">
<h2>SR Data Prep</h2>
<pre class="r"><code>##### Libraries
library(dplyr)
library(tidyr)

### Function: create tibble from data path
antiNull &lt;- function(csvPath){
  
  #Read in table
  allRaw &lt;- tibble(read.csv(csvPath))
  
  #filter for non-null values
  starveRaw &lt;- allRaw %&gt;% select(line,starvation) %&gt;% filter(!is.na(starvation))
  
  return(starveRaw)
}

#Female data
csvPathF &lt;- &quot;/data/morgante_lab/data/dgrp/phenotypes/eQTL_traits_females.csv&quot;
starveF &lt;- antiNull(csvPathF)

#Male data
csvPathM &lt;- &quot;/data/morgante_lab/data/dgrp/phenotypes/eQTL_traits_males.csv&quot;
starveM &lt;- antiNull(csvPathM)

#rename columns for combination
colnames(starveF) &lt;- c(&quot;line&quot;, &quot;f&quot;)
colnames(starveM) &lt;- c(&quot;line&quot;, &quot;m&quot;)

#Combine data
starve &lt;- starveF %&gt;% mutate(m = starveM$m)

#compute difference and average
starve &lt;- starve %&gt;% select(line, f, m) %&gt;% mutate(dif = f - m, avg = (f+m)/2)

#Sort by difference, cosmetic
starve &lt;- arrange(starve, starve$dif)

#Save, colnames are lin,f,m,dif,avg
write.csv(starve, &quot;/data/morgante_lab/nklimko/rep/dgrp-starve/data/starve.csv&quot;)</code></pre>
</div>
<div id="sr-analysis" class="section level2">
<h2>SR Analysis</h2>
<pre class="r"><code>#Libraries and setup
library(dplyr)
library(data.table)
library(tidyr)
starve &lt;- tibble(read.csv(&quot;/data/morgante_lab/nklimko/rep/dgrp-starve/data/starve.csv&quot;)) %&gt;% select(-X)
x &lt;- starve$f
y &lt;- starve$m

#plot parameters
par(mfrow=c(1,2))

#Female Histogram
hist(x, 
     col=&quot;red&quot;,
     border=&quot;black&quot;,
     prob=TRUE,
     xlab = &quot;Starvation Resistance&quot;,
     main = &quot;Female Lines&quot;)

lines(density(x),
      lwd = 2,
      col = &quot;black&quot;)

#Male Histogram
hist(y, 
     col=&quot;blue&quot;,
     border=&quot;black&quot;,
     prob=TRUE,
     xlab = &quot;Starvation Resistance&quot;,
     main = &quot;Male Lines&quot;)

lines(density(y),
      lwd = 2,
      col = &quot;black&quot;)

#Summary stats
cbind(summary(x), summary(y))


# Comparative Boxplot
boxplot(x,
        y,
        col = c(&quot;red&quot;,&quot;blue&quot;),
        names=c(&quot;Female&quot;, &quot;Male&quot;),
        main=&quot;Sex Comparison of Starvation Resistance&quot;,
        xlab=&quot;Starvation Resistance&quot;,
        ylab=&quot;Sex&quot;,
        horizontal = TRUE)

# Scatter Plot
plot(x,
     y,
     col=&quot;black&quot;,
     xlab=&quot;Female Starvation Resistance&quot;,
     ylab=&quot;Male Starvation Resistance&quot;,
     abline(reg=lm(y~x), 
            col=&quot;purple&quot;))
text(x = 85, y = 25,
     &quot;y = 14.2347x + 0.5192&quot;,
     cex = 0.75)
text(x = 85, y = 22.5,
     &quot;R=0.4693, p-value &lt; 2.2e-16&quot;,
     cex = 0.75)

#Trendline parameter determination
summary(lm(y~x))

# QQ Plots
#plot parameters
par(mfrow=c(1,2))

qqnorm(x, main=&quot;Female Distribution&quot;)
qqline(x)
qqnorm(y, main=&quot;Male Distribution&quot;)
qqline(y)

#Shapiro-Wilk Normality Tests
Female_Starvation &lt;- x
Male_Starvation &lt;- y

shapiro.test(Female_Starvation)
shapiro.test(Male_Starvation)


### Group selection

#plot parameters
par(mfrow=c(1,2))

#Difference
hist(starve$dif, 
     col=&quot;purple&quot;,
     border=&quot;black&quot;,
     xlab = &quot;Change in Starvation Resistance&quot;,
     main = &quot;Difference of Female and Male Lines&quot;)

#Average
hist(starve$avg, 
     col=&quot;pink&quot;,
     border=&quot;black&quot;,
     xlab = &quot;Average Starvation Resistance&quot;,
     main = &quot;Avg of Female and Male Lines&quot;)

### Scatter Plot
plot(starve$avg,
     starve$dif,
     col=&quot;black&quot;,
     xlab=&quot;Line Average&quot;,
     ylab=&quot;Intersex Difference&quot;,
     abline(reg=lm(starve$dif~starve$avg), 
            col=&quot;purple&quot;))
text(x = 72, y = -12.5,
     &quot;y = 0.3258 - 2.3984&quot;,
     cex = 0.75)
text(x = 72, y = -15,
     &quot;R=0.1293, p-value &lt; 1.37e-7&quot;,
     cex = 0.75)

#Trendline parameter determination
summary(lm(starve$dif~starve$avg))</code></pre>
</div>
<div id="ingroup-prep" class="section level2">
<h2>Ingroup Prep</h2>
<pre class="r"><code>##### Libraries
library(dplyr)
library(tidyr)
library(data.table)

#Read in data
starve &lt;- tibble(read.csv(&quot;data/starve.csv&quot;)) %&gt;% select(-X)

#Manual inspection of data to create cutoffs for ingroups
print(arrange(starve, starve$dif), n=205)
print(arrange(starve, starve$avg), n=205)

# Lines with male SR high than female SR
difMinus &lt;- starve %&gt;% filter(dif &lt; 0) %&gt;% arrange(line) %&gt;% select(line)
difMinus &lt;- as.data.table(difMinus)
fwrite(difMinus, &quot;../data/difMinus.txt&quot;)

# Lines with largest gap from male SR to female SR
difPlus &lt;- starve %&gt;% filter(dif &gt;= 30) %&gt;% arrange(line) %&gt;% select(line)
difPlus &lt;- as.data.table(difPlus)
fwrite(difPlus, &quot;../data/difPlus.txt&quot;)

# Lowest average SR between males and females
avgMinus &lt;- starve %&gt;% filter(avg &lt; 40) %&gt;% arrange(line) %&gt;% select(line)
avgMinus &lt;- as.data.table(avgMinus)
fwrite(avgMinus, &quot;../data/avgMinus.txt&quot;)

# Highest average SR between males and females
avgPlus &lt;- starve %&gt;% filter(dif &gt;= 30) %&gt;% arrange(line) %&gt;% select(line)
avgPlus &lt;- as.data.table(avgPlus)
fwrite(avgPlus, &quot;../data/avgPlus.txt&quot;)</code></pre>
</div>
<div id="snp-gathering" class="section level2">
<h2>SNP Gathering</h2>
<pre class="r"><code>### Libraries
library(&quot;data.table&quot;)
library(&quot;dplyr&quot;)

##change per job
dataPath &lt;- &quot;/data/morgante_lab/data/dgrp/genotypes/dgrp2_tgeno_filtered_meanimputed.txt&quot;
targetPath &lt;- &quot;/data/morgante_lab/nklimko/rep/dgrp-starve/data/INPUT.txt&quot;
finalPath &lt;- &quot;/data/morgante_lab/nklimko/rep/dgrp-starve/data/CHANGE.txt&quot;

#number of results to save
finalCount &lt;- 200

#files read in
data &lt;- fread(dataPath) 
inGroup &lt;- fread(targetPath)

#modification of inGroup to vector type, clunky
inGroup &lt;- inGroup[,line]

#IDs saved and removed
var_id &lt;- data[,var_id]
data &lt;- data[, var_id:=NULL]

#data partitioned by column
inData &lt;- data[,colnames(data) %in% inGroup, with=FALSE]
outData &lt;- data[,!(colnames(data) %in% inGroup), with=FALSE]

#means calculated for every row
inMean &lt;- rowMeans(inData)
outMean &lt;- rowMeans(outData)
  
#table constructed of marker IDs and group means
phaseA &lt;- data.table(var_id, inMean, outMean)

#calculates difference of group means
trueData &lt;-  phaseA[, result:=inMean - outMean]
#trueData &lt;-  phaseA[, result:=inMean / outMean]

#orders data by absolute value of difference, largest first
trueSort &lt;- trueData[order(-abs(result))]

#index controls number of top results to save
finalSet&lt;- trueSort[1:finalCount]

#writes top (finalCount) to designated path
fwrite(finalSet, finalPath)</code></pre>
</div>
<div id="snp-to-gene" class="section level2">
<h2>SNP to Gene</h2>
<pre class="r"><code>### Libraries
library(data.table)
library(dplyr)
library(stringr)

#input paths
genePath &lt;- &quot;/data/databases/flybase/fb-r5.57/fb-r5.57.clean.gene.bound&quot;
snpPath &lt;- &quot;/data/morgante_lab/nklimko/rep/dgrp-starve/data/snpList.txt&quot;

#read data in
genBank &lt;- fread(genePath, col.names = c(&quot;leg&quot;,&quot;start&quot;, &quot;stop&quot;, &quot;strand&quot;,&quot;gene&quot;))
snpList &lt;- fread(snpPath)

#split SNP tag to search gene data correctly
snpList[, arm := tstrsplit(var_id, split=&quot;_&quot;,fixed = TRUE)[1]]
snpList[, pos := tstrsplit(var_id, split=&quot;_&quot;,fixed = TRUE)[2]]
snpList[, pos := as.numeric(pos)]

#set start and stop to numeric type
genBank[, &#39;:=&#39;(start=as.numeric(start), stop=as.numeric(stop))]

#extract gene where position contained and arm matches
geneHits &lt;- genBank[snpList, on = .(start &lt;= pos, stop &gt;= pos,leg==arm), .(ori, geneFind = gene)][!is.na(geneFind)]

#save file
fwrite(geneHits, &quot;/data/morgante_lab/nklimko/rep/dgrp-starve/data/geneHits.txt&quot;)</code></pre>
</div>
<div id="gene-to-go" class="section level2">
<h2>Gene to GO</h2>
<pre class="r"><code>### Libraries
library(data.table)
library(dplyr)

#input paths
goPath &lt;- &quot;/data/databases/flybase/fb-r5.57/gene_go.table&quot;
genePath &lt;- &quot;/data/morgante_lab/nklimko/rep/dgrp-starve/data/geneHits.txt&quot;

#read data in
genList &lt;- fread(genePath, col.names = c(&quot;ori&quot;, &quot;gene&quot;))
goBank &lt;- fread(goPath, col.names = c(&quot;gene&quot;, &quot;mf&quot;, &quot;bp&quot;, &quot;cc&quot;))

#match go terms to gene
genList &lt;- goBank[genList, on = .(gene)]

#save
fwrite(genList, &quot;./goGroups.txt&quot;)

#No origin
table(genList[,mf])
table(genList[,cc])
table(genList[,bp])

##This one
table(genList[,gene, by=ori])
table(genList[,mf, by=ori])
table(genList[,cc, by=ori])
table(genList[,bp, by=ori])</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /opt/ohpc/pub/Software/openblas_0.3.10/lib/libopenblas_haswellp-r0.3.10.dev.so

locale:
 [1] LC_CTYPE=en_US.utf-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.utf-8        LC_COLLATE=en_US.utf-8    
 [5] LC_MONETARY=en_US.utf-8    LC_MESSAGES=en_US.utf-8   
 [7] LC_PAPER=en_US.utf-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.utf-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.7.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8.3     highr_0.9        bslib_0.3.1      jquerylib_0.1.4 
 [5] compiler_4.0.3   pillar_1.7.0     later_1.3.0      git2r_0.30.1    
 [9] tools_4.0.3      getPass_0.2-2    digest_0.6.29    jsonlite_1.8.0  
[13] evaluate_0.15    tibble_3.1.6     lifecycle_1.0.1  pkgconfig_2.0.3 
[17] rlang_1.0.4      cli_3.3.0        rstudioapi_0.13  yaml_2.3.5      
[21] xfun_0.30        fastmap_1.1.0    httr_1.4.2       stringr_1.4.0   
[25] knitr_1.38       sass_0.4.1       fs_1.5.2         vctrs_0.4.1     
[29] rprojroot_2.0.3  glue_1.6.2       R6_2.5.1         processx_3.5.3  
[33] fansi_1.0.3      rmarkdown_2.16   callr_3.7.0      magrittr_2.0.3  
[37] whisker_0.4      ps_1.6.0         promises_1.2.0.1 htmltools_0.5.2 
[41] ellipsis_0.3.2   httpuv_1.6.5     utf8_1.2.2       stringi_1.7.6   
[45] crayon_1.5.1    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
