<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Starvation Resistance Analysis</title>

<script src="site_libs/header-attrs-2.16/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">dgrp-starve</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="starve.html">Analysis</a>
</li>
<li>
  <a href="scripts.html">Scripts</a>
</li>
<li>
  <a href="multiComp.html">Multiple Comparison</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/nklimko/dgrp-starve">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Starvation Resistance Analysis</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-11-22
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>dgrp-starve/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221101code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221101)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221101code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221101)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/data/morgante_lab/nklimko/rep/dgrp-starve/data/INPUT.txt
</td>
<td style="text-align:left;">
data/INPUT.txt
</td>
</tr>
<tr>
<td style="text-align:left;">
/data/morgante_lab/nklimko/rep/dgrp-starve/data/CHANGE.txt
</td>
<td style="text-align:left;">
data/CHANGE.txt
</td>
</tr>
<tr>
<td style="text-align:left;">
/data/morgante_lab/nklimko/rep/dgrp-starve/data/snpList.txt
</td>
<td style="text-align:left;">
data/snpList.txt
</td>
</tr>
<tr>
<td style="text-align:left;">
/data/morgante_lab/nklimko/rep/dgrp-starve/data/geneHits.txt
</td>
<td style="text-align:left;">
data/geneHits.txt
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomnklimkodgrpstarvetreee82e73c2fe383dcb0cc7a3f70564bc6ba061e23ctargetblanke82e73ca">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/nklimko/dgrp-starve/tree/e82e73c2fe383dcb0cc7a3f70564bc6ba061e23c" target="_blank">e82e73c</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomnklimkodgrpstarvetreee82e73c2fe383dcb0cc7a3f70564bc6ba061e23ctargetblanke82e73ca"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/nklimko/dgrp-starve/tree/e82e73c2fe383dcb0cc7a3f70564bc6ba061e23c" target="_blank">e82e73c</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  code/analysisSR.R
    Untracked:  code/geneGO.R
    Untracked:  code/multiPrep.R
    Untracked:  code/snpGene.77509.err
    Untracked:  code/snpGene.77509.out
    Untracked:  code/snpGene.77515.err
    Untracked:  code/snpGene.77515.out
    Untracked:  code/snpGene.sbatch
    Untracked:  data/eQTL_traits_females.csv
    Untracked:  data/eQTL_traits_males.csv
    Untracked:  data/goGroups.txt
    Untracked:  data/xp-f.txt
    Untracked:  data/xp-m.txt
    Untracked:  scoreAnalysisMulticomp.R
    Untracked:  temp.Rmd

Unstaged changes:
    Deleted:    analysis/database.Rmd
    Modified:   code/baseScript-lineComp.R
    Modified:   code/fourLinePrep.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/starve.Rmd</code>) and HTML
(<code>docs/starve.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nklimko/dgrp-starve/f09779ec190667ec4bd3e7d1b1dd24f717eb9970/docs/starve.html" target="_blank">f09779e</a>
</td>
<td>
nklimko
</td>
<td>
2022-11-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nklimko/dgrp-starve/f1d6437766ad9ecd416ac54f034cbad7f7bdcf9d/docs/starve.html" target="_blank">f1d6437</a>
</td>
<td>
nklimko
</td>
<td>
2022-11-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nklimko/dgrp-starve/79e98ca129310232141bbab69da3f597907100ae/docs/starve.html" target="_blank">79e98ca</a>
</td>
<td>
nklimko
</td>
<td>
2022-11-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nklimko/dgrp-starve/d23d7f71489e3ed50d081bccd668c65a499071cc/docs/starve.html" target="_blank">d23d7f7</a>
</td>
<td>
nklimko
</td>
<td>
2022-11-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/3a282f1db3d21899f3e0d61f62defef61b7de01c/analysis/starve.Rmd" target="_blank">3a282f1</a>
</td>
<td>
nklimko
</td>
<td>
2022-11-11
</td>
<td>
wflow_publish("analysis/*")
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/0ce7be48447410eb66c9a868dd7003092cbfa2c8/analysis/starve.Rmd" target="_blank">0ce7be4</a>
</td>
<td>
nklimko
</td>
<td>
2022-11-10
</td>
<td>
code sync
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nklimko/dgrp-starve/0ce7be48447410eb66c9a868dd7003092cbfa2c8/docs/starve.html" target="_blank">0ce7be4</a>
</td>
<td>
nklimko
</td>
<td>
2022-11-10
</td>
<td>
code sync
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/b90e54038b40f0c360dc050a8c726c5e7ed34081/analysis/starve.Rmd" target="_blank">b90e540</a>
</td>
<td>
nklimko
</td>
<td>
2022-11-09
</td>
<td>
wflow_git_commit(all = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nklimko/dgrp-starve/b90e54038b40f0c360dc050a8c726c5e7ed34081/docs/starve.html" target="_blank">b90e540</a>
</td>
<td>
nklimko
</td>
<td>
2022-11-09
</td>
<td>
wflow_git_commit(all = TRUE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/81037cc126bd17fa21accfe4362f79110103ffc8/analysis/starve.Rmd" target="_blank">81037cc</a>
</td>
<td>
nklimko
</td>
<td>
2022-11-09
</td>
<td>
html reworking, snp success
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nklimko/dgrp-starve/81037cc126bd17fa21accfe4362f79110103ffc8/docs/starve.html" target="_blank">81037cc</a>
</td>
<td>
nklimko
</td>
<td>
2022-11-09
</td>
<td>
html reworking, snp success
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/41de4a88be30888f7f8343b90d36afae626d0792/analysis/starve.Rmd" target="_blank">41de4a8</a>
</td>
<td>
nklimko
</td>
<td>
2022-11-08
</td>
<td>
difMinus success, reframe for all four genotype analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nklimko/dgrp-starve/41de4a88be30888f7f8343b90d36afae626d0792/docs/starve.html" target="_blank">41de4a8</a>
</td>
<td>
nklimko
</td>
<td>
2022-11-08
</td>
<td>
difMinus success, reframe for all four genotype analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/d2abea6feeda54688be2af3b6c738cdf570d1cb4/analysis/starve.Rmd" target="_blank">d2abea6</a>
</td>
<td>
nklimko
</td>
<td>
2022-11-05
</td>
<td>
11/5 Phase plan, data prep/analysis nested steps and pseudo for script
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nklimko/dgrp-starve/d2abea6feeda54688be2af3b6c738cdf570d1cb4/docs/starve.html" target="_blank">d2abea6</a>
</td>
<td>
nklimko
</td>
<td>
2022-11-05
</td>
<td>
11/5 Phase plan, data prep/analysis nested steps and pseudo for script
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nklimko/dgrp-starve/blob/d0a5cb116cf12fd2c174a8863630523808a12675/analysis/starve.Rmd" target="_blank">d0a5cb1</a>
</td>
<td>
nklimko
</td>
<td>
2022-11-05
</td>
<td>
Baseline analysis 11/5, horrendous formatting
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="about" class="section level1">
<h1>About</h1>
<p>The lesser fruit fly, <em>Drosophila melanogaster</em>, is a model
oganism due to short life cycles, small genomes that allow targetted
manipulation, and analogous organ systems to humans. This allows for an
accelerated rate of genetic testing and manipulation while bypassing
major ethical concerns of experimenting on humans.</p>
<p>Extensive data collection has been performed on behavioral, fitness,
and morphological traits for both male and female lines. In this
project, I looked at starvation analysis between male and female lines
in the DGRP.</p>
</div>
<div id="starvation-resistance-data-prep" class="section level1">
<h1>Starvation Resistance Data Prep</h1>
<p>The first step in this project was to determine which lines could be
used for comparison. Not all lines have experimental data for every
trait, so all lines with NA for starvation resistance had to be excluded
as no comparisons could be made. After doing this, I created a table
with line ID, male resistance, and female resistance. On top of this, I
calculated the average resistance and the change in resistance between
sexes and recorded these in additional columns.</p>
</div>
<div id="analysis-of-starvation-resistance" class="section level1">
<h1>Analysis of Starvation Resistance</h1>
<p>In order to get a better understanding of what I was working with, I
plotted the data using various methods.</p>
<div id="histograms-of-starvation-resistance" class="section level3">
<h3>Histograms of Starvation Resistance</h3>
<p><img src="figure/starve.Rmd/2-base-hist-1.svg" width="672" style="display: block; margin: auto;" /></p>
<p>Plotting histograms of each sex reveals both are unimodal with mild
skew to the right. This skew demonstrates he effect of selective
pressures as low starvation resistance decreases overall fitness of the
fly.</p>
</div>
<div id="comparative-boxplot" class="section level3">
<h3>Comparative Boxplot</h3>
<p><img src="figure/starve.Rmd/2-boxplot-1.svg" width="672" style="display: block; margin: auto;" /></p>
<p>Plotting the distributions on the same axis gives a more accurate
representation of distribution between sexes. The average female has a
much greater starvation resistance of 60.665 compared to the average
male 45.732. Outliers to the right of both boxplots confirm right skew
seen in the histograms.</p>
</div>
<div id="scatter-plot" class="section level3">
<h3>Scatter Plot</h3>
<p><img src="figure/starve.Rmd/2-fm-scatter-1.svg" width="672" style="display: block; margin: auto;" /></p>
<p>The trendline is a simple linear regression with R and p values
extracted from summary statistics. The slope indicates a positive
correlation between male and female starvation resistance while the R
value of 0.4693 indicates that the correlation is not strong.</p>
</div>
<div id="qq-plots" class="section level3">
<h3>QQ Plots</h3>
<p><img src="figure/starve.Rmd/2-qqplot-1.svg" width="672" style="display: block; margin: auto;" /></p>
<p>There is some systematic deviation from linearity on both tails in
females and on the upper tail in males. In order to more accurately
assess normality, I found the Shapiro-Wilk Test for normality and chose
his method due to its retention of power and prior implementation in
R.</p>
</div>
<div id="shapiro-wilk-tests-for-normality" class="section level3">
<h3>Shapiro-Wilk Tests for Normality</h3>
<p>The test statistic formula for the Shapiro-Wilk test is: <span
class="math display">\[
W = \frac{(\sum_{i=1}^{n}a_ix_{(i)})^{2}}{\sum_{i=1}^{n}(x_i-\overline
x)^{2}}
\]</span></p>
<p>More about the Shapiro-Wilk test statistic can be found <a
href="https://en.wikipedia.org/wiki/Shapiro%E2%80%93Wilk_test#Theory">here</a>.
The closer this statistic is to 1 indicates how normal a distribution
is, with 1 being perfectly normal.</p>
<pre><code>
    Shapiro-Wilk normality test

data:  Female_Starvation
W = 0.97662, p-value = 0.001826</code></pre>
<pre><code>
    Shapiro-Wilk normality test

data:  Male_Starvation
W = 0.98745, p-value = 0.07023</code></pre>
<p>The W values indicate that both samples are mostly normal with
slightly more deviation in females than in males.</p>
<p>At an <span class="math inline">\(\alpha\)</span> level of 0.05, we
have sufficient evidence to reject that the Female population is
normally distributed as p &lt; <span
class="math inline">\(\alpha\)</span> (0.001826 &lt; 0.05).</p>
<p>At an <span class="math inline">\(\alpha\)</span> level of 0.05, we
do not have sufficient evidence to reject that the Male population is
normally distributed as p &gt; <span
class="math inline">\(\alpha\)</span> (0.07023 &gt; 0.05). Notably the
correlation is not strong as the p-value is close to 0.05.</p>
</div>
</div>
<div id="ingroup-preparation" class="section level1">
<h1>Ingroup Preparation</h1>
<p>In order to dig durther into starvation resistance, I started looking
for possible trends between difference in starvation resistance and
average resistance per line.</p>
<div id="histograms" class="section level3">
<h3>Histograms</h3>
<p><img src="figure/starve.Rmd/2-dif-avg-hist-1.svg" width="672" style="display: block; margin: auto;" />
The histograms of each reveal another unimodal distribution. What I did
not expect was to see any negative values in line difference. These
lines had males with higher starvation resistance than females.</p>
</div>
<div id="scatter-plot-1" class="section level3">
<h3>Scatter Plot</h3>
<p><img src="figure/starve.Rmd/2-dif-avg-scatter-1.svg" width="672" style="display: block; margin: auto;" /></p>
<p>There is almost no correlation between the traits as R is 0.1293 for
the trendline. This plot was mostly for practice with scatter plots and
linear regression.</p>
<p>The histograms, however, led in another direction. I figured that
looking at the extremities of differences and averages was correlated to
some genetic difference. I began by extracting the line IDs of each
group of interest(inGroups) to separae files to run further
analyses.</p>
<p>Example line extraction for the negative male difference is shown
here:</p>
<pre class="r"><code># Lines with male SR high than female SR
difMinus &lt;- starve %&gt;% filter(dif &lt; 0) %&gt;% arrange(line) %&gt;% select(line)
difMinus &lt;- as.data.table(difMinus)
fwrite(difMinus, &quot;./data/difMinus.txt&quot;)</code></pre>
<p>The four groups of interest created were: - avgMinus: Lines with low
average starvation resistance - avgPlus: Lines with high average
starvation resistance - difMinus: Lines with higher male than female
starvation resistance - difPlus: Lines with higher female than male
starvation resistance</p>
</div>
</div>
<div id="snp-extraction" class="section level1">
<h1>SNP Extraction</h1>
<p>With these groups, I wanted to look at SNP variation in the ingroup
and compare SNP prevalence to the remainder of the lines. I reasoned
that any locus that varies dramatically between ingroup and outgroup may
contribute to starvation resistance.</p>
<p>The math behind this operation looked at the average SNP value for
each group at each locus and copmuted the difference. High
differences(closer to 2) indicate that a SNP existed in one group at
much higher rates than the other. The data was then ordered by absolute
value to catch cases where the ingroup had an snp across lines and where
all lines had an untouched region.</p>
<p>This template was used on each ingroup to find SNPs of interest using
the methodology above.</p>
<pre class="r"><code>##change per job
dataPath &lt;- &quot;/data/morgante_lab/data/dgrp/genotypes/dgrp2_tgeno_filtered_meanimputed.txt&quot;
targetPath &lt;- &quot;/data/morgante_lab/nklimko/rep/dgrp-starve/data/INPUT.txt&quot;
finalPath &lt;- &quot;/data/morgante_lab/nklimko/rep/dgrp-starve/data/CHANGE.txt&quot;

#number of results to save
finalCount &lt;- 200

#files read in
data &lt;- fread(dataPath) 
inGroup &lt;- fread(targetPath)

#modification of inGroup to vector type, clunky
inGroup &lt;- inGroup[,line]

#IDs saved and removed
var_id &lt;- data[,var_id]
data &lt;- data[, var_id:=NULL]

#data partitioned by column
inData &lt;- data[,colnames(data) %in% inGroup, with=FALSE]
outData &lt;- data[,!(colnames(data) %in% inGroup), with=FALSE]

#means calculated for every row
inMean &lt;- rowMeans(inData)
outMean &lt;- rowMeans(outData)
  
#table constructed of marker IDs and group means
phaseA &lt;- data.table(var_id, inMean, outMean)

#calculates difference of group means
trueData &lt;-  phaseA[, result:=inMean - outMean]

#orders data by absolute value of difference, largest first
trueSort &lt;- trueData[order(-abs(result))]

#index controls number of top results to save
finalSet&lt;- trueSort[1:finalCount]

#writes top (finalCount) to designated path
fwrite(finalSet, finalPath)</code></pre>
</div>
<div id="snp-curation" class="section level1">
<h1>SNP Curation</h1>
<p>After this job was run, I compiled all four groups into a single data
sheet and keyed them to the original group they had come from to
streamline data analysis.</p>
<pre class="r"><code># Set max number of results
index &lt;- 50

#read in &quot;index&quot; number of results, set above
dM &lt;- fread(&quot;../data/difMinus-result.txt&quot;, nrows = index)
dP &lt;- fread(&quot;../data/difPlus-result.txt&quot;, nrows = index)
aM &lt;- fread(&quot;../data/avgMinus-result.txt&quot;, nrows = index)
aP &lt;- fread(&quot;../data/avgPlus-result.txt&quot;, nrows = index)

#mark origin of SNP
dM[,ori:=&quot;difMinus&quot;]
dP[,ori:=&quot;difPlus&quot;]
aM[,ori:=&quot;avgMinus&quot;]
aP[,ori:=&quot;avgPlus&quot;]

#combine tables
snpList &lt;- rbind(dM, dP, aM, aP)

#remove data columns by assigning to NULL
snpList[,&#39;:=&#39;(inMean=NULL, outMean=NULL, result=NULL)]

#Save file
fwrite(snpList, &quot;/data/morgante_lab/nklimko/rep/dgrp-starve/data/snpList.txt&quot;)</code></pre>
</div>
<div id="gene-extraction" class="section level1">
<h1>Gene Extraction</h1>
<p>With this list of SNPs, I wanted to match the SNPs to their location
in the <em>D. melanogaster</em> genome. This was done by joining the SNP
compilation table to the Flybase gene boundary table in order to extract
gene codes in the FlyBase gene format(FBgn0000000).</p>
<pre class="r"><code>### Libraries
library(data.table)
library(dplyr)
library(stringr)

#input paths
genePath &lt;- &quot;/data/databases/flybase/fb-r5.57/fb-r5.57.clean.gene.bound&quot;
snpPath &lt;- &quot;/data/morgante_lab/nklimko/rep/dgrp-starve/data/snpList.txt&quot;

#read data in
genBank &lt;- fread(genePath, col.names = c(&quot;leg&quot;,&quot;start&quot;, &quot;stop&quot;, &quot;strand&quot;,&quot;gene&quot;))
snpList &lt;- fread(snpPath)

#split SNP tag to search gene data correctly
snpList[, arm := tstrsplit(var_id, split=&quot;_&quot;,fixed = TRUE)[1]]
snpList[, pos := tstrsplit(var_id, split=&quot;_&quot;,fixed = TRUE)[2]]
snpList[, pos := as.numeric(pos)]

#set start and stop to numeric type
genBank[, &#39;:=&#39;(start=as.numeric(start), stop=as.numeric(stop))]

#extract gene where position contained and arm matches
geneHits &lt;- genBank[snpList, on = .(start &lt;= pos, stop &gt;= pos,leg==arm), .(ori, geneFind = gene)][!is.na(geneFind)]

#write file
fwrite(geneHits, &quot;/data/morgante_lab/nklimko/rep/dgrp-starve/data/geneHits.txt&quot;)</code></pre>
</div>
<div id="go-term-extraction" class="section level1">
<h1>GO Term Extraction</h1>
<p>For my final data.table, I matched extracted genes to their
corresponding GO terms in the flybase gene_go.table file. From there, I
summarized the GO terms present within each group to determine what
processes or structures were most affected that could potentially cause
observable differences in starvation resistance between lines. <a
href="http://amigo.geneontology.org/amigo/medial_search">amiGO
browser</a></p>
<p>Extracted GO terms using code below</p>
<pre class="r7"><code>#No origin
table(genList[,mf])
table(genList[,cc])
table(genList[,bp])

##This one
table(genList[,gene, by=ori])
table(genList[,mf, by=ori])
table(genList[,cc, by=ori])
table(genList[,bp, by=ori])</code></pre>
<div id="future-directions" class="section level2">
<h2>Future Directions</h2>
<p>Fix GO Term Extraction.</p>
<p>Theoretically, this analysis could also be applied to gene expression
instead of SNPs as an initial step and follow through with the same
material. Gene expression and SNP data could be compared to see which
functions are affected by both or only one group</p>
<p>On a different track, multiple regression using other traits in the
eQTL csv would be a logical next step. Probably should have attempted
this with one of the sugar traits.</p>
</div>
<div id="reflection" class="section level2">
<h2>Reflection</h2>
<p>Altogether, I realized that I strayed from a purely statistical
analysis to more of a GWAS. That aside, this project has been good
practice for using data.table and Rmarkdown. Most of the initial
scripting and some of the later scripts are hard-coded to these examples
which is not great is I wanted to extend this analysis to other
traits.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /opt/ohpc/pub/Software/openblas_0.3.10/lib/libopenblas_haswellp-r0.3.10.dev.so

locale:
 [1] LC_CTYPE=en_US.utf-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.utf-8        LC_COLLATE=en_US.utf-8    
 [5] LC_MONETARY=en_US.utf-8    LC_MESSAGES=en_US.utf-8   
 [7] LC_PAPER=en_US.utf-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.utf-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] tidyr_1.2.0       stringr_1.4.0     data.table_1.14.2 dplyr_1.0.8      
[5] workflowr_1.7.0  

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8.3     highr_0.9        bslib_0.3.1      jquerylib_0.1.4 
 [5] compiler_4.0.3   pillar_1.7.0     later_1.3.0      git2r_0.30.1    
 [9] tools_4.0.3      getPass_0.2-2    digest_0.6.29    jsonlite_1.8.0  
[13] evaluate_0.15    tibble_3.1.6     lifecycle_1.0.1  pkgconfig_2.0.3 
[17] rlang_1.0.4      DBI_1.1.2        cli_3.3.0        rstudioapi_0.13 
[21] yaml_2.3.5       xfun_0.30        fastmap_1.1.0    httr_1.4.2      
[25] knitr_1.38       generics_0.1.2   sass_0.4.1       fs_1.5.2        
[29] vctrs_0.4.1      tidyselect_1.1.2 rprojroot_2.0.3  glue_1.6.2      
[33] R6_2.5.1         processx_3.5.3   fansi_1.0.3      rmarkdown_2.16  
[37] purrr_0.3.4      callr_3.7.0      magrittr_2.0.3   whisker_0.4     
[41] ps_1.6.0         promises_1.2.0.1 htmltools_0.5.2  ellipsis_0.3.2  
[45] assertthat_0.2.1 httpuv_1.6.5     utf8_1.2.2       stringi_1.7.6   
[49] crayon_1.5.1    </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
