<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Iterative Comparison</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">dgrp-starve</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/nklimko/dgrp-starve">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Iterative Comparison</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-02-20
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>dgrp-starve/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div class="panel-group" id="workflowr-checks">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221101code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20221101)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221101code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221101)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/data/morgante_lab/nklimko/rep/dgrp-starve/data/iterFold.txt
</td>
<td style="text-align:left;">
data/iterFold.txt
</td>
</tr>
<tr>
<td style="text-align:left;">
/data/morgante_lab/nklimko/rep/dgrp-starve/data/fMeans.txt
</td>
<td style="text-align:left;">
data/fMeans.txt
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomnklimkodgrpstarvetree5d140d6f248941da26e56e1719035dc403aeed4etargetblank5d140d6a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/nklimko/dgrp-starve/tree/5d140d6f248941da26e56e1719035dc403aeed4e" target="_blank">5d140d6</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomnklimkodgrpstarvetree5d140d6f248941da26e56e1719035dc403aeed4etargetblank5d140d6a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/nklimko/dgrp-starve/tree/5d140d6f248941da26e56e1719035dc403aeed4e" target="_blank">5d140d6</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData

Untracked files:
    Untracked:  analysis/linearReg.Rmd
    Untracked:  analysis/stepwise.Rmd
    Untracked:  code/aaaTest
    Untracked:  code/analysisSR.R
    Untracked:  code/geneGO.R
    Untracked:  code/multiPrep.R
    Untracked:  code/regress.81916.err
    Untracked:  code/regress.81916.out
    Untracked:  code/regress.81918.err
    Untracked:  code/regress.81918.out
    Untracked:  code/regress.R
    Untracked:  code/regress.sbatch
    Untracked:  code/regressF.81919.err
    Untracked:  code/regressF.81919.out
    Untracked:  code/regressF.R
    Untracked:  code/regressF.sbatch
    Untracked:  code/regress_f_adj.109973.err
    Untracked:  code/regress_f_adj.109973.out
    Untracked:  code/regress_f_adj.109974.err
    Untracked:  code/regress_f_adj.109974.out
    Untracked:  code/regress_f_adj.R
    Untracked:  code/regress_f_adj.sbatch
    Untracked:  code/regress_m_adj.109971.err
    Untracked:  code/regress_m_adj.109971.out
    Untracked:  code/regress_m_adj.109972.err
    Untracked:  code/regress_m_adj.109972.out
    Untracked:  code/regress_m_adj.R
    Untracked:  code/regress_m_adj.sbatch
    Untracked:  code/snpGene.77509.err
    Untracked:  code/snpGene.77509.out
    Untracked:  code/snpGene.77515.err
    Untracked:  code/snpGene.77515.out
    Untracked:  code/snpGene.sbatch
    Untracked:  data/eQTL_traits_females.csv
    Untracked:  data/eQTL_traits_males.csv
    Untracked:  data/fRegress.txt
    Untracked:  data/fRegress_adj.txt
    Untracked:  data/f_adj.txt
    Untracked:  data/goGroups.txt
    Untracked:  data/mPart.txt
    Untracked:  data/mRegress.txt
    Untracked:  data/mRegress_adj.txt
    Untracked:  data/m_adj.txt
    Untracked:  data/multiReg.rData
    Untracked:  data/starve-f.txt
    Untracked:  data/starve-m.txt
    Untracked:  data/xp-f.txt
    Untracked:  data/xp-m.txt
    Untracked:  data/xpOLDf
    Untracked:  data/xpOLDm-xp-m.txt
    Untracked:  data/y_save.txt
    Untracked:  figure/
    Untracked:  lmm.R
    Untracked:  qqdum.R
    Untracked:  scoreAnalysisMulticomp.R
    Untracked:  temp.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>#loop count and data limit
iter &lt;- 8

#Parallel core count
registerDoParallel(cores = 10)

print(paste0(&quot;Iteration count: &quot;, iter))</code></pre>
<pre><code>[1] &quot;Iteration count: 8&quot;</code></pre>
<pre class="r"><code>#ggplot holder list
gg &lt;- vector(mode=&#39;list&#39;, length=12)

# result storage elements
fit_greml &lt;- vector(mode=&#39;list&#39;, length=iter)
fit_gbayesC &lt;- vector(mode=&#39;list&#39;, length=iter)
fit_varbvs &lt;- vector(mode=&#39;list&#39;, length=iter)
fit_glmnet &lt;- vector(mode=&#39;list&#39;, length=iter)

#wolb infection and inversion status data with phenotype adjustment function
load(&quot;/data/morgante_lab/data/dgrp/misc/adjustData.RData&quot;)

#expression data matched to line and starvation phenotype
xp_f &lt;- fread(&quot;data/xp-f.txt&quot;)
xp_m &lt;- fread(&quot;data/xp-m.txt&quot;)</code></pre>
<pre class="r"><code>dataFlag &lt;- TRUE

if(dataFlag){

 
  
  #expression data matched to line and starvation phenotype
  xp_f &lt;- fread(&quot;data/xp-f.txt&quot;)
  
  xp_m &lt;- fread(&quot;data/xp-m.txt&quot;)
  
  #setwd(&quot;C:/Users/noahk/OneDrive/Desktop/amogus&quot;)
  #getwd()
  
  #create matrix of only gene expression, trims line and starvation
  X &lt;- as.matrix(xp_f[,3:11340])
  rownames(X) &lt;- xp_f[,line]
  W &lt;- scale(X)
  
  y_temp &lt;- xp_f[,starvation]
  dat &lt;- data.frame(id=xp_f[,line], y=y_temp)
  y &lt;- adjustPheno(dat, &quot;starvation&quot;)

} else{
  
  # Toy Data set, 200x100 matrix
  W &lt;- matrix(rnorm(20000), ncol = 100)
  colnames(W) &lt;- paste0(&quot;gene&quot;, 1:ncol(W))
  rownames(W) &lt;- paste0(&quot;line&quot;, 1:nrow(W))
  
  #model uses genes 1:5 and 10:20
  y &lt;- rowSums(W[, 1:5]) + rowSums(W[, 10:20]) + rnorm(nrow(W))

}</code></pre>
<pre><code>Type III ANOVA table for covariates: starvation
                 Df Sum of Sq   RSS    AIC F value  Pr(&gt;F)  
&lt;none&gt;                        28767 1009.8                  
factor(wolba)     1    483.10 29250 1011.1  3.1236 0.07881 .
factor(In_2L_t)   2     60.22 28827 1006.2  0.1947 0.82325  
factor(In_2R_NS)  2    300.22 29067 1007.8  0.9706 0.38078  
factor(In_3R_P)   2     85.92 28853 1006.4  0.2778 0.75777  
factor(In_3R_K)   2    959.43 29726 1012.3  3.1017 0.04731 *
factor(In_3R_Mo)  2    416.65 29184 1008.6  1.3470 0.26255  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1


Estimated effects
                    Estimate Std. Error    t value     Pr(&gt;|t|)
(Intercept)       58.7465186   1.553929 37.8051442 3.103130e-89
factor(wolba)y     3.2450739   1.836101  1.7673724 7.880567e-02
factor(In_2L_t)1  -0.8506378   3.101713 -0.2742477 7.841986e-01
factor(In_2L_t)2  -1.8583916   3.151661 -0.5896547 5.561378e-01
factor(In_2R_NS)1  0.8630221   4.621570  0.1867379 8.520697e-01
factor(In_2R_NS)2  6.7057598   4.825270  1.3897169 1.662760e-01
factor(In_3R_P)1  -1.8513843   5.206148 -0.3556150 7.225319e-01
factor(In_3R_P)2   4.0596759   6.331413  0.6411959 5.221847e-01
factor(In_3R_K)1   7.4870256   4.084124  1.8332024 6.837092e-02
factor(In_3R_K)2  15.5762633   8.948152  1.7407240 8.338564e-02
factor(In_3R_Mo)1 -5.4712766   4.279569 -1.2784644 2.026787e-01
factor(In_3R_Mo)2 -3.6074434   3.224295 -1.1188318 2.646547e-01</code></pre>
<pre class="r"><code>### qgg_greml

#model to solve for, vector of ones
mu &lt;- matrix(rep(1, length(y)), ncol=1)
#names(mu) &lt;- paste0(&quot;line&quot;, 1:length(mu))
rownames(mu) &lt;- xp_f[,line]
TRM &lt;- tcrossprod(W)/ncol(W)

# k-fold parameters
n &lt;- length(y)
fold &lt;- 10</code></pre>
<pre class="r"><code>for(i in 1:iter) {

  
  jk &lt;- 0

}</code></pre>
<p>###2023/02/15</p>
<p>Fit four models to real expression data for both genders Email Vijay/Maria about secretariat access or use OoD to rename files Email Dr. Starr-Moss(cc) about admission to graduate program in the Fall Varbvs uses a spike-slab prior, glmnet LASSO variable selection</p>
<pre class="r"><code>#this runs on the same data sets as greml

### qgg_gBayesC
### https://www.rdocumentation.org/packages/qgg/versions/1.1.1/topics/gbayes

# Bayes C: uses a “rounded spike” (low-variance Gaussian) at origin many small effects can contribute to polygenic component, reduces the dimensionality of the model (makes Gibbs sampling feasible).

#Parallel Header
#tempResult &lt;- 


corLoop &lt;- foreach(i=1:iter) %dopar% {

#Linear Header
#for(i in 1:iter){
  
  corResult &lt;- (1:4)
  
  
  
  #setup train and test sets with trait vectors
  test_IDs &lt;- sample(1:n, as.integer(n / fold))
  
  W_train &lt;- W[-test_IDs,]
  W_test &lt;- W[test_IDs,]
  y_train &lt;- y[-test_IDs]
  y_test &lt;- y[test_IDs]
  
  
  ### GREML, qgg package
  fitGreml &lt;- qgg::greml(y=y, X=mu, GRM=list(A=TRM), validate = matrix(test_IDs,ncol=1), verbose=FALSE)
  
  #Store coeff directly
  fit_greml[[i]] &lt;- fitGreml$accuracy$Corr
  
  corResult[1] &lt;- fitGreml$accuracy$Corr

  
  
  ### GBAYES-C
  fitC &lt;- qgg::gbayes(y=y_train, W=W_train, method=&quot;bayesC&quot;, scaleY=FALSE)
  
  # expected/calculated value for y_test
  # \hat{y}_test = W_{test} * \hat{b} + \hat{mu}
  y_calc &lt;- W_test %*% fitC$b + mean(y_train)
  
  # store coeff
  fit_gbayesC[[i]] &lt;- cor(y_test, y_calc)
  
  corResult[2] &lt;- cor(y_test, y_calc)
  
  ### VARBVS
  fitVarb &lt;- varbvs::varbvs(X = W_train, NULL, y=y_train, family = &quot;gaussian&quot;, logodds=seq(-3.5,-1,0.1), sa = 1, verbose=FALSE)
  
  # \hat{y}_test = W_{test} * \hat{b} + \hat{mu}
  y_calc &lt;- W_test %*% fitVarb$beta + mean(y_train)
  
  fit_varbvs[[i]] &lt;- cor(y_test, y_calc)
  
  corResult[3] &lt;- cor(y_test, y_calc)
  
  
  
  
  ### GLMNET  
  fitlm &lt;- glmnet::cv.glmnet(x=W_train, y=y_train, alpha=1)
  

  b_hat &lt;- glmnet::coef.glmnet(fitlm, s=&quot;lambda.min&quot;)


  y_int &lt;- b_hat[1]

  b_hat &lt;- b_hat[2:length(b_hat)]

  y_calc &lt;- W_test %*% b_hat + y_int

  fit_glmnet[[i]] &lt;- cor(y_test, y_calc)
  
  corResult[4] &lt;- cor(y_test, y_calc)
  
  
  corResult

}</code></pre>
<pre class="r"><code>for(i in 1:iter){
  fit_greml[[i]] &lt;- corLoop[[i]][1]
  fit_gbayesC[[i]] &lt;- corLoop[[i]][2]
  fit_varbvs[[i]] &lt;- corLoop[[i]][3]
  fit_glmnet[[i]] &lt;- corLoop[[i]][4]
  
}</code></pre>
<pre class="r"><code>bint &lt;- iter/10

### qgg_greml

qgg_greml_data &lt;- as.data.table(fit_greml)
qgg_greml_data &lt;- transpose(qgg_greml_data)
colnames(qgg_greml_data) &lt;- &quot;cor&quot;

#print(paste0(&quot;GREML&quot;))
#print(paste0(&quot;Mean correlation coefficient: &quot;, mean(qgg_greml_data[,cor])))
#print(paste0(&quot;Variance of correlation coefficient: &quot;, var(qgg_greml_data)))

gg[[1]] &lt;- ggplot(qgg_greml_data, aes(x=cor)) +
  geom_histogram(bins=bint, fill=&#39;red&#39;) +
  labs(x=&quot;Corr Coeff&quot;) +
  ggtitle(&quot;GREML CV Correlations Histogram&quot;)


### qgg_gbayesC

qgg_gbayesC_data &lt;- as.data.table(fit_gbayesC)
qgg_gbayesC_data &lt;- transpose(qgg_gbayesC_data)
colnames(qgg_gbayesC_data) &lt;- &quot;cor&quot;

#print(paste0(&quot;gBayesC&quot;))
#print(paste0(&quot;Mean correlation coefficient: &quot;, mean(qgg_gbayesC_data[,cor])))
#print(paste0(&quot;Variance of correlation coefficient: &quot;, var(qgg_gbayesC_data)))

gg[[2]] &lt;- ggplot(qgg_gbayesC_data, aes(x=cor)) +
  geom_histogram(bins=bint, fill=&#39;red&#39;) +
  labs(x=&quot;Corr Coeff&quot;) +
  ggtitle(&quot;gBayesC Prediction Corr Histogram&quot;)


### VARBVS

varbvs_data &lt;- as.data.table(fit_varbvs)
varbvs_data &lt;- transpose(varbvs_data)
colnames(varbvs_data) &lt;- &quot;cor&quot;

#print(paste0(&quot;VARBVS&quot;))
#print(paste0(&quot;Mean correlation coefficient: &quot;, mean(varbvs_data[,cor])))
#print(paste0(&quot;Variance of correlation coefficient: &quot;, var(varbvs_data)))

gg[[3]] &lt;- ggplot(varbvs_data, aes(x=cor)) +
  geom_histogram(bins=bint, fill=&#39;red&#39;) +
  labs(x=&quot;Corr Coeff&quot;) +
  ggtitle(&quot;varbvs Prediction Corr Histogram&quot;)


### GLMNET

glmnet_data &lt;- as.data.table(fit_glmnet)
glmnet_data &lt;- transpose(glmnet_data)
colnames(glmnet_data) &lt;- &quot;cor&quot;

#print(paste0(&quot;GLMNET&quot;))
#print(paste0(&quot;Mean correlation coefficient: &quot;, mean(glmnet_data[,cor])))
#print(paste0(&quot;Variance of correlation coefficient: &quot;, var(glmnet_data)))

gg[[4]] &lt;- ggplot(glmnet_data, aes(x=cor)) +
  geom_histogram(bins=bint, fill=&#39;red&#39;) +
  labs(x=&quot;Corr Coeff&quot;) +
  ggtitle(&quot;glmnet Prediction Corr Histogram&quot;)</code></pre>
<pre class="r"><code>result &lt;- data.table(method=&quot;greml&quot;, meanCoeff=mean(qgg_greml_data[,cor])) 
result[,varCoeff := var(qgg_greml_data)]
                    
temp &lt;- data.table(method=&quot;gBayesC&quot;, meanCoeff=mean(qgg_gbayesC_data[,cor]))
temp[,varCoeff := var(qgg_gbayesC_data)]
result &lt;- rbind(result, temp)

temp &lt;- data.table(method=&quot;varbvs&quot;, meanCoeff=mean(varbvs_data[,cor]))
temp[,varCoeff := var(varbvs_data)]
result &lt;- rbind(result, temp)

temp &lt;- data.table(method=&quot;glmnet&quot;, meanCoeff=mean(glmnet_data[,cor]))
temp[,varCoeff := var(glmnet_data)]
result &lt;- rbind(result, temp)

# Map the time of day to different fill colors
#gg[[5]] &lt;- ggplot(data=result, aes(x=method, y=meanCoeff, fill=method)) +
    #geom_bar(stat=&quot;identity&quot;)</code></pre>
<pre class="r"><code>plot_grid(gg[[1]],gg[[2]], ncol=2)</code></pre>
<pre><code>Warning: Computation failed in `stat_bin()`:
Need at least one bin.
Computation failed in `stat_bin()`:
Need at least one bin.</code></pre>
<p><img src="figure/stepwise.Rmd/plot-print-1.svg" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_grid(gg[[3]],gg[[4]], ncol=2)</code></pre>
<pre><code>Warning: Computation failed in `stat_bin()`:
Need at least one bin.
Computation failed in `stat_bin()`:
Need at least one bin.</code></pre>
<p><img src="figure/stepwise.Rmd/plot-print-2.svg" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>library(doParallel)
registerDoParallel(cores = 4)


# file path for final table
filePath &lt;- &quot;/data/morgante_lab/nklimko/rep/dgrp-starve/data/iterFold.txt&quot;

# gene expression data from earlier bound to line and starvation
fMeans &lt;- fread(&quot;/data/morgante_lab/nklimko/rep/dgrp-starve/data/fMeans.txt&quot;)
start &lt;- 3
end &lt;- 11340
#f 11340
#m 13577

# foreach creates a list with index indicating position in loop
# function grabs the p-value from correlation of starvation to expression of trait
pvalList &lt;- foreach(i=start:end) %dopar% {  
  temp &lt;- summary(lm(fMeans[,c(2,i), with=FALSE], na.action=na.omit))[[4]][8]  
}


# converts list to vector
part &lt;- unlist(pvalList, use.names = FALSE)

# binds p-vals to column index 
id &lt;- start:end
part &lt;- data.table(id,pvalList)


# write table to file
fwrite(part, filePath)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /opt/ohpc/pub/Software/openblas_0.3.10/lib/libopenblas_haswellp-r0.3.10.dev.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] glmnet_4.1-6      Matrix_1.5-3      varbvs_2.5-16     qgg_1.1.1        
 [5] doParallel_1.0.17 iterators_1.0.14  foreach_1.5.2     qqman_0.1.8      
 [9] cowplot_1.1.1     ggplot2_3.3.5     data.table_1.14.2 dplyr_1.0.8      

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8.3        lattice_0.20-45     png_0.1-7          
 [4] assertthat_0.2.1    rprojroot_2.0.3     digest_0.6.29      
 [7] utf8_1.2.2          R6_2.5.1            MatrixModels_0.5-1 
[10] evaluate_0.15       coda_0.19-4         highr_0.9          
[13] pillar_1.7.0        rlang_1.0.4         rstudioapi_0.13    
[16] SparseM_1.81        jquerylib_0.1.4     rmarkdown_2.16     
[19] splines_4.0.3       statmod_1.4.37      stringr_1.4.0      
[22] munsell_0.5.0       compiler_4.0.3      httpuv_1.6.5       
[25] xfun_0.30           pkgconfig_2.0.3     shape_1.4.6        
[28] mcmc_0.9-7          htmltools_0.5.2     tidyselect_1.1.2   
[31] tibble_3.1.6        workflowr_1.7.0     codetools_0.2-18   
[34] fansi_1.0.3         calibrate_1.7.7     crayon_1.5.1       
[37] withr_2.5.0         later_1.3.0         MASS_7.3-56        
[40] grid_4.0.3          jsonlite_1.8.0      gtable_0.3.0       
[43] lifecycle_1.0.1     DBI_1.1.2           git2r_0.30.1       
[46] magrittr_2.0.3      scales_1.2.0        cli_3.3.0          
[49] stringi_1.7.6       fs_1.5.2            promises_1.2.0.1   
[52] latticeExtra_0.6-29 bslib_0.3.1         ellipsis_0.3.2     
[55] generics_0.1.2      vctrs_0.4.1         nor1mix_1.3-0      
[58] RColorBrewer_1.1-3  tools_4.0.3         glue_1.6.2         
[61] purrr_0.3.4         jpeg_0.1-9          survival_3.3-1     
[64] fastmap_1.1.0       yaml_2.3.5          colorspace_2.0-3   
[67] knitr_1.38          sass_0.4.1          quantreg_5.94      
[70] MCMCpack_1.6-3     </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
