---
title: "Linear Mixed Model"
output:
  workflowr::wflow_html:
    toc: true
    latex_engine: "xelatex"
    code_folding: "hide"
editor_options:
  chunk_output_type: console
---
  
```{r 0-setup, include=FALSE, warning=FALSE}
library(dplyr)
library(data.table)
library(ggplot2)
library(cowplot)
library(qqman)
library(qgg)

set.seed(1)

options(bitmapType = "cairo")

```

# Methodology

The lowest p-values were chosen to represent the fixed effect as better correlations assist in prediction of starvation resistance.

Random effect was a normal distribution with a mean of 1 and sd of 0.25. Default norm resulted in errors with negative value affecting matrix computations or creating singular matrices.

Additionally, I left verbose on TRUE to observe iteration steps as while randomness is seeded, iteration steps were not the same every time.


Depending on what is run, the system is not able to converge as it either goes to positive infinity and overflows to a negative value or reaches zero and "converges" at zero. If this webpage renders, then both runs will have converged on zero.

Note: data shown in this instance is wrong as highest p-values were chosen as the prediction almost always converged to zero, allowing compilation with workflowr 

# Female 

```{r f-qq}

#read in expression data
fMeans <- fread("data/fMeans.txt")

#create matrix of only gene expression, trims line and starvation
X <- as.matrix(fMeans[,3:11340])
rownames(X) <- fMeans[,line]

y <- fMeans[,starvation]

###Create model for covariates to adjust for (only an intercept in our case)
mu <- rep(1, length(y))

###Compute transcriptomic relationship matrix (accounts for structure based on expression levels)
W <- scale(X)
TRM <- tcrossprod(W)/ncol(W)

###Fit mixed model
fit <- greml(y = y, X = mu, GRM = list(TRM), verbose = TRUE, maxit = 100)

stat <- glma(fit = fit, W = W)

#qqplot
qq(stat[,4], main="Female Gene p-values")


```

# Male 

```{r m-qq}

rm(list=ls())
set.seed(1)

#read in expression data
mMeans <- fread("data/mMeans.txt")

#create matrix of only gene expression, trims line and starvation
X <- as.matrix(mMeans[,3:13577])
rownames(X) <- mMeans[,line]

y <- mMeans[,starvation]

###Create model for covariates to adjust for (only an intercept in our case)
mu <- rep(1, length(y))

###Compute transcriptomic relationship matrix (accounts for structure based on expression levels)
W <- scale(X)
TRM <- tcrossprod(W)/ncol(W)

###Fit mixed model
fit <- greml(y = y, X = mu, GRM = list(TRM), verbose = TRUE, maxit = 100)

stat <- glma(fit = fit, W = W)

#qqplot
qq(stat[,4], main="Male Gene p-values")


```



