---
title: "Linear Mixed Model"
output:
  workflowr::wflow_html:
    toc: true
    latex_engine: "xelatex"
    code_folding: "hide"
editor_options:
  chunk_output_type: console
---
  
```{r 0-setup, include=FALSE, warning=FALSE}
library(dplyr)
library(data.table)
library(ggplot2)
library(cowplot)
library(qqman)
library(qgg)

set.seed(1)

options(bitmapType = "cairo")

```

# Methodology

Multiple regression is the process of determining the coefficients of a model with each coefficient corresponding to the effect of one explanatory variable. 

With Genomic Reduced Maximum Likelihood, or GREML, a design matrix is calculated using a Genomic Relationship matrix and a vector of 1s to find the intercepts. 

Multiplying the vector for starvation resistance by the inverse of the design matrix solves for the coefficients of the multivariate model. The residuals must also be subtracted and consitute the random effect of the mixed model.


For each data set, the expression data matrix(line x gene) and the starvation resistance vector(line x 1) were taken from prior lab data. From the [PCA Project](pca.html), a dataframe containing line, starvation resistancec, and expression data per gene was repurposed for this model.

The Transcriptomic Relation Matrix was calculated using the cross product of a scaled expression matrix and multiplied by the inverse of the number of columns. 


# Female 

```{r f-qq, warning=FALSE}

#read in expression data
fMeans <- fread("data/fMeans.txt")

#create matrix of only gene expression, trims line and starvation
X <- as.matrix(fMeans[,3:11340])
rownames(X) <- fMeans[,line]

y <- fMeans[,starvation]

###Create model for covariates to adjust for (only an intercept in our case)
mu <- rep(1, length(y))

###Compute transcriptomic relationship matrix (accounts for structure based on expression levels)
W <- scale(X)
TRM <- tcrossprod(W)/ncol(W)

###Fit mixed model
fit <- greml(y = y, X = mu, GRM = list(TRM), verbose = TRUE, maxit = 100)

stat <- glma(fit = fit, W = W)

#qqplot
qq(stat[,4], main="Female Gene p-values")


```

# Male 

```{r m-qq, warning=FALSE}

#read in expression data
mMeans <- fread("data/mMeans.txt")

#create matrix of only gene expression, trims line and starvation
X <- as.matrix(mMeans[,3:13577])
rownames(X) <- mMeans[,line]

y <- mMeans[,starvation]

###Create model for covariates to adjust for (only an intercept in our case)
mu <- rep(1, length(y))

###Compute transcriptomic relationship matrix (accounts for structure based on expression levels)
W <- scale(X)
TRM <- tcrossprod(W)/ncol(W)

###Fit mixed model
fit <- greml(y = y, X = mu, GRM = list(TRM), verbose = TRUE, maxit = 100)

stat <- glma(fit = fit, W = W)

#qqplot
qq(stat[,4], main="Male Gene p-values")

```


Multiple regression of both female and male structures show effects of the genetic architecture on the models as both systematically deviate from linearity. 

[QQ Plots from Linear Regression](linReg.html#QQ_Plots)




