---
title: "Female Multiple Regression and Prediction"
output:
  workflowr::wflow_html:
    toc: true
    latex_engine: "xelatex"
    code_folding: "hide"
editor_options:
  chunk_output_type: console
---


```{r 0-setup, include=FALSE, warning=FALSE}
library(dplyr)
library(data.table)
library(ggplot2)
library(cowplot)
library(qqman)
library(qgg)

set.seed(1)

options(bitmapType = "cairo")
options(error = function() traceback(3))

```

# Methodology

Multiple regression is the process of determining the coefficients of a model with each coefficient corresponding to the effect of one explanatory variable. 

With Genomic Reduced Maximum Likelihood, or GREML, a design matrix is calculated using a Genomic Relationship matrix and a vector of 1s to find the intercepts. 

Multiplying the vector for starvation resistance by the inverse of the design matrix solves for the coefficients of the multivariate model. The residuals must also be subtracted and consitute the random effect of the mixed model.

For each data set, the expression data matrix(line x gene) and the starvation resistance vector(line x 1) were taken from prior lab data. From the [PCA Project](pca.html), a dataframe containing line, starvation resistance, and expression data per gene was repurposed for this model.

The Transcriptomic Relation Matrix was calculated using the cross product of a scaled expression matrix and multiplied by the inverse of the number of columns. 

```{r f-multiReg}

#wolb infection and inversion status data with phenotype adjustment function
load("/data/morgante_lab/data/dgrp/misc/adjustData.RData")

#read in expression data
fMeans <- fread("data/fMeans.txt")

#create matrix of only gene expression, trims line and starvation
X <- as.matrix(fMeans[,3:11340])
rownames(X) <- fMeans[,line]

#extract and adjust phenotype(starvation)
y <- fMeans[,starvation]
dat <- data.frame(id=fMeans[,line], y=y)
y_adj <- adjustPheno(dat, "starvation")

#scale matrix and compute TRM using crossproduct and number of markers(genes)
W <- scale(X)
TRM <- tcrossprod(W)/ncol(W)

#convert TRM structure to list
listTRM <- list(A=TRM)

#model to solve for, vector of ones
mu <- matrix(rep(1, length(y_adj)), ncol=1)

# REML analyses
fitG <- greml(y = y_adj, X = mu, GRM = listTRM, verbose = TRUE)

#general linear model analysis
statTemp <- glma(fit = fitG, W = W)

#summary(cvTB$accuracy$Corr)
par(mfrow=c(1,2))

#histogram of coefficients
hist(statTemp[,1], main="Female Coefficients")

#qq plot of p-values
qq(statTemp[,4], main="Female Gene p-values")

statF <- statTemp


```



```{r f-predict}

# k-fold parameters
n <- length(y_adj)
fold <- 10
nvalid <- 50

#validate set creation
validate <- replicate(nvalid, sample(1:n, as.integer(n / fold)))

#cross-validation greml
cvTB <- greml(y = y_adj, X = mu, GRM = listTRM, validate = validate, verbose=FALSE)

#summary statistics of correlation
#summary(cvTB$accuracy$Corr)

gg <- vector(mode='list', length=6)

histData <- data.table(cor = cvTB$accuracy$Corr)

#female mean
gg[[1]] <- ggplot(histData, aes(x=cor)) +
  geom_histogram(bins=8, fill='red') +
  labs(x="CV Trial", y="Correlation Coefficient") +
  ggtitle("Female CV Correlations")

gg[[1]]


sd(cvTB$accuracy$Corr)
sd(cvTB$accuracy$MSPE)

```

The mean Correlation coefficient for all trials was `r mean(cvTB$accuracy$Corr)` with variance `r var(cvTB$accuracy$Corr)`.

The mean of Mean Square Predicted Error for all trials was `r mean(cvTB$accuracy$MSPE)` with variance `r var(cvTB$accuracy$MSPE)`.






