---
title: "Starvation Resistance Analysis"
output:
  workflowr::wflow_html:
    toc: true
    latex_engine: "xelatex"
editor_options:
  chunk_output_type: console
---

```{r 0-setup, include=FALSE}
library(dplyr)
library(data.table)
library(stringr)
library(tidyr)
starve <- tibble(read.csv("data/starve.csv")) %>% select(-X)
x <- starve$f
y <- starve$m

```

# About

The lesser fruit fly, *Drosophila melanogaster*, is a model oganism due to short life cycles, small genomes that allow targetted manipulation, and analogous organ systems to humans. This allows for an accelerated rate of genetic testing and manipulation while bypassing major ethical concerns of experimenting on humans. 

Extensive data collection has been performed on behavioral, fitness, and morphological traits for both male and female lines. In this project, I looked at starvation analysis between male and female lines in the DGRP.

# Starvation Resistance Data Prep

The first step in this project was to determine which lines could be used for comparison. Not all lines have experimental data for every trait, so all lines with NA for starvation resistance had to be excluded as no comparisons could be made. After doing this, I created a table with line ID, male resistance, and female resistance. On top of this, I calculated the average resistance and the change in resistance between sexes and recorded these in additional columns.

# Analysis of Starvation Resistance

In order to get a better understanding of what I was working with, I plotted the data using various methods. 

### Histograms of Starvation Resistance
```{r 2-base-hist, echo=FALSE}

par(mfrow=c(1,2))

#Female
hist(x, 
     col="red",
     border="black",
     prob=TRUE,
     xlab = "Starvation Resistance",
     main = "Female Lines")

lines(density(x),
      lwd = 2,
      col = "black")

#Male
hist(y, 
     col="blue",
     border="black",
     prob=TRUE,
     xlab = "Starvation Resistance",
     main = "Male Lines")

lines(density(y),
      lwd = 2,
      col = "black")

```

Plotting histograms of each sex reveals both are unimodal with mild skew to the right. This skew demonstrates he effect of selective pressures as low starvation resistance decreases overall fitness of the fly. 

### Comparative Boxplot
```{r 2-boxplot, echo = FALSE}
boxplot(x,
        y,
        col = c("red","blue"),
        names=c("Female", "Male"),
        main="Sex Comparison of Starvation Resistance",
        xlab="Starvation Resistance",
        ylab="Sex",
        horizontal = TRUE)

```

Plotting the distributions on the same axis gives a more accurate representation of distribution between sexes. The average female has a much greater starvation resistance of `r round(mean(x), digits=3)` compared to the average male `r round(mean(y), digits=3)`. Outliers to the right of both boxplots confirm right skew seen in the histograms. 

### Scatter Plot
```{r 2-fm-scatter, echo = FALSE}
plot(x,
     y,
     col="black",
     xlab="Female Starvation Resistance",
     ylab="Male Starvation Resistance",
     abline(reg=lm(y~x), 
            col="purple"))
text(x = 85, y = 25,
     "y = 14.2347x + 0.5192",
     cex = 0.75)
text(x = 85, y = 22.5,
     "R=0.4693, p-value < 2.2e-16",
     cex = 0.75)

#summary(lm(y~x))

```

The trendline is a simple linear regression with R and p values extracted from summary statistics. The slope indicates a positive correlation between male and female starvation resistance while the R value of 0.4693 indicates that the correlation is not strong. 

### QQ Plots
```{r 2-qqplot, echo = FALSE}
par(mfrow=c(1,2))

qqnorm(x, main="Female Distribution")
qqline(x)
qqnorm(y, main="Male Distribution")
qqline(y)


```

There is some systematic deviation from linearity on both tails in females and on the upper tail in males. In order to more accurately assess normality, I found the Shapiro-Wilk Test for normality and chose his method due to its retention of power and prior implementation in R.

### Shapiro-Wilk Tests for Normality

The test statistic formula for the Shapiro-Wilk test is:
$$
W = \frac{(\sum_{i=1}^{n}a_ix_{(i)})^{2}}{\sum_{i=1}^{n}(x_i-\overline x)^{2}}
$$

More about the Shapiro-Wilk test statistic can be found [here](https://en.wikipedia.org/wiki/Shapiro%E2%80%93Wilk_test#Theory). The closer this statistic is to 1 indicates how normal a distribution is, with 1 being perfectly normal.

```{r 2-shapiro, echo = FALSE}

Female_Starvation <- x
Male_Starvation <- y

shapiro.test(Female_Starvation)
shapiro.test(Male_Starvation)
```

The W values indicate that both samples are mostly normal with slightly more deviation in females than in males.

At an $\alpha$ level of 0.05, we have sufficient evidence to reject that the Female population is normally distributed as p < $\alpha$ (0.001826 < 0.05).

At an $\alpha$ level of 0.05, we do not have sufficient evidence to reject that the Male population is normally distributed as p > $\alpha$ (0.07023 > 0.05). Notably the correlation is not strong as the p-value is close to 0.05.


# Ingroup Preparation

In order to dig durther into starvation resistance, I started looking for possible trends between difference in starvation resistance and average resistance per line. 

### Histograms
```{r 2-dif-avg-hist, echo = FALSE}

par(mfrow=c(1,2))

#Difference
hist(starve$dif, 
     col="purple",
     border="black",
     xlab = "Change in Starvation Resistance",
     main = "Difference of Female/Male Lines")

#Average
hist(starve$avg, 
     col="pink",
     border="black",
     xlab = "Average Starvation Resistance",
     main = "Avg of Female/Male Lines")

```
The histograms of each reveal another unimodal distribution. What I did not expect was to see any negative values in line difference. These lines had males with higher starvation resistance than females. 

### Scatter Plot
```{r 2-dif-avg-scatter, echo = FALSE}
plot(starve$avg,
     starve$dif,
     col="black",
     xlab="Line Average",
     ylab="Intersex Difference",
     abline(reg=lm(starve$dif~starve$avg), 
            col="purple"))
text(x = 72, y = -12.5,
     "y = 0.3258 - 2.3984",
     cex = 0.75)
text(x = 72, y = -15,
     "R=0.1293, p-value < 1.37e-7",
     cex = 0.75)

#summary(lm(starve$dif~starve$avg))

```

There is almost no correlation between the traits as R is 0.1293 for the trendline. This plot was mostly for practice with scatter plots and linear regression.

The histograms, however, led in another direction. I figured that looking at the extremities of differences and averages was correlated to some genetic difference. I began by extracting the line IDs of each group of interest(inGroups) to separae files to run further analyses.

Example line extraction for the negative male difference is shown here:

```{r 3-ingroup, eval=FALSE} 
# Lines with male SR high than female SR
difMinus <- starve %>% filter(dif < 0) %>% arrange(line) %>% select(line)
difMinus <- as.data.table(difMinus)
fwrite(difMinus, "./data/difMinus.txt")
```

The four groups of interest created were:
- avgMinus: Lines with low average starvation resistance
- avgPlus: Lines with high average starvation resistance
- difMinus: Lines with higher male than female starvation resistance
- difPlus: Lines with higher female than male starvation resistance


# SNP Extraction

With these groups, I wanted to look at SNP variation in the ingroup and compare SNP prevalence to the remainder of the lines. I reasoned that any locus that varies dramatically between ingroup and outgroup may contribute to starvation resistance.

The math behind this operation looked at the average SNP value for each group at each locus and copmuted the difference. High differences(closer to 2) indicate that a SNP existed in one group at much higher rates than the other. The data was then ordered by absolute value to catch cases where the ingroup had an snp across lines and where all lines had an untouched region.  

This template was used on each ingroup to find SNPs of interest using the methodology above.

```{r 4-snpExtraction, eval=FALSE}

##change per job
dataPath <- "/data/morgante_lab/data/dgrp/genotypes/dgrp2_tgeno_filtered_meanimputed.txt"
targetPath <- "/data/morgante_lab/nklimko/rep/dgrp-starve/data/INPUT.txt"
finalPath <- "/data/morgante_lab/nklimko/rep/dgrp-starve/data/CHANGE.txt"

#number of results to save
finalCount <- 200

#files read in
data <- fread(dataPath) 
inGroup <- fread(targetPath)

#modification of inGroup to vector type, clunky
inGroup <- inGroup[,line]

#IDs saved and removed
var_id <- data[,var_id]
data <- data[, var_id:=NULL]

#data partitioned by column
inData <- data[,colnames(data) %in% inGroup, with=FALSE]
outData <- data[,!(colnames(data) %in% inGroup), with=FALSE]

#means calculated for every row
inMean <- rowMeans(inData)
outMean <- rowMeans(outData)
  
#table constructed of marker IDs and group means
phaseA <- data.table(var_id, inMean, outMean)

#calculates difference of group means
trueData <-  phaseA[, result:=inMean - outMean]

#orders data by absolute value of difference, largest first
trueSort <- trueData[order(-abs(result))]

#index controls number of top results to save
finalSet<- trueSort[1:finalCount]

#writes top (finalCount) to designated path
fwrite(finalSet, finalPath)
```

# SNP Curation

After this job was run, I compiled all four groups into a single data sheet and keyed them to the original group they had come from to streamline data analysis.

```{r 5-snp, eval=FALSE}
# Set max number of results
index <- 50

#read in "index" number of results, set above
dM <- fread("../data/difMinus-result.txt", nrows = index)
dP <- fread("../data/difPlus-result.txt", nrows = index)
aM <- fread("../data/avgMinus-result.txt", nrows = index)
aP <- fread("../data/avgPlus-result.txt", nrows = index)

#mark origin of SNP
dM[,ori:="difMinus"]
dP[,ori:="difPlus"]
aM[,ori:="avgMinus"]
aP[,ori:="avgPlus"]

#combine tables
snpList <- rbind(dM, dP, aM, aP)

#remove data columns by assigning to NULL
snpList[,':='(inMean=NULL, outMean=NULL, result=NULL)]

#Save file
fwrite(snpList, "/data/morgante_lab/nklimko/rep/dgrp-starve/data/snpList.txt")
```

# Gene Extraction

With this list of SNPs, I wanted to match the SNPs to their location in the *D. melanogaster* genome. This was done by joining the SNP compilation table to the Flybase gene boundary table in order to extract gene codes in the FlyBase gene format(FBgn0000000).

```{r 6-geneExtract, eval=FALSE}
### Libraries
library(data.table)
library(dplyr)
library(stringr)

#input paths
genePath <- "/data/databases/flybase/fb-r5.57/fb-r5.57.clean.gene.bound"
snpPath <- "/data/morgante_lab/nklimko/rep/dgrp-starve/data/snpList.txt"

#read data in
genBank <- fread(genePath, col.names = c("leg","start", "stop", "strand","gene"))
snpList <- fread(snpPath)

#split SNP tag to search gene data correctly
snpList[, arm := tstrsplit(var_id, split="_",fixed = TRUE)[1]]
snpList[, pos := tstrsplit(var_id, split="_",fixed = TRUE)[2]]
snpList[, pos := as.numeric(pos)]

#set start and stop to numeric type
genBank[, ':='(start=as.numeric(start), stop=as.numeric(stop))]

#extract gene where position contained and arm matches
geneHits <- genBank[snpList, on = .(start <= pos, stop >= pos,leg==arm), .(ori, geneFind = gene)][!is.na(geneFind)]

#write file
fwrite(geneHits, "/data/morgante_lab/nklimko/rep/dgrp-starve/data/geneHits.txt")
```

# GO Term Extraction
For my final data.table, I matched extracted genes to their corresponding GO terms in the flybase gene_go.table file. From there, I summarized the GO terms present within each group to determine what processes or structures were most affected that could potentially cause observable differences in starvation resistance between lines. [amiGO browser](http://amigo.geneontology.org/amigo/medial_search)

Extracted GO terms using code below

```{r7, eval=FALSE}
#No origin
table(genList[,mf])
table(genList[,cc])
table(genList[,bp])

##This one
table(genList[,gene, by=ori])
table(genList[,mf, by=ori])
table(genList[,cc, by=ori])
table(genList[,bp, by=ori])
```

## Future Directions

Fix GO Term Extraction. 

Theoretically, this analysis could also be applied to gene expression instead of SNPs as an initial step and follow through with the same material. Gene expression and SNP data could be compared to see which functions are affected by both or only one group

On a different track, multiple regression using other traits in the eQTL csv would be a logical next step. Probably should have attempted this with one of the sugar traits.

## Reflection
Altogether, I realized that I strayed from a purely statistical analysis to more of a GWAS. That aside, this project has been good practice for using data.table and Rmarkdown. Most of the initial scripting and some of the later scripts are hard-coded to these examples which is not great is I wanted to extend this analysis to other traits. 
